---
header-includes:
- \usepackage{graphicx,latexsym}
- \usepackage{amssymb,amsthm,amsmath}
- \usepackage{longtable,booktabs,setspace}
---

<!--- Actinos-->
<!--
You can delete the header-includes (lines 3-6 above) if you like and also the chunk below since it is loaded in the skeleton.Rmd file.  They are included so that chap3.Rmd will compile by itself when you hit Knit PDF.
-->

```{r include_packages_4, include = FALSE}
# This chunk ensures that the reedtemplates package is
# installed and loaded. This reedtemplates package includes
# the template files for the thesis and also two functions
# used for labeling and referencing
if(!require(devtools))
  install.packages("devtools", repos = "http://cran.rstudio.com")
if(!require(dplyr))
    install.packages("dplyr", repos = "http://cran.rstudio.com")
if(!require(ggplot2))
    install.packages("ggplot2", repos = "http://cran.rstudio.com")
if(!require(reedtemplates)){
  library(devtools)
  devtools::install_github("ismayc/reedtemplates")
  }
library(reedtemplates)
flights <- read.csv("data/flights.csv")
```



# Actinobacteria EvoMining Results {#ref_labels}
Actinobacteria is an ancient phylum {Referencia de luis}  

## Tables

----------------------------------------------------------------------------------
  Factors                    Correlation between Parents & Child     
------------------------- ----------------------------------------- --------------
  GenomeDB                                  1245                         
  
  Families                                  65                          
------------------------- ----------------------------------------- --------------
Table: Correlation of Inheritance Factors for Parents and Child \label{tab:inher}


```{r loading_tables_Actino,message=FALSE,echo=FALSE}
library(ggplot2)
library( reshape )
library(plyr )
library(RColorBrewer)
library(scales)
library(plyr)
library(RColorBrewer)
hm.palette <- colorRampPalette(rev(brewer.pal(11, 'Spectral')), space='Lab')  

#setwd("/home/nelly/GIT/phd/Thesis/tesis/chap5_files")
ActinosCentral <- read.table("chapter4/ActinoCentral", header=TRUE, sep="\t")
ActinosHeatPlot <- read.table("chapter4/ActinoHeatPlot", header=TRUE, sep="\t")
ActinosNp <- read.table("chapter4/ActinoNp", header=TRUE, sep="\t")
ActinosSMASH <- read.table("chapter4/ActinosSmashResults", header=TRUE, sep="\t")
ActinosTaxa <- read.table("chapter4/ActinoTaxa", header=TRUE, sep="\t")


```


```{r Actinosumary,echo=FALSE}
#summary(ActinosTaxa)
#table(ActinosTaxa$Order)
```



\clearpage
<!-- clearpage ends the page, and also dumps out all floats.Floats are things like tables and figures. 
-->
<!--
Plan general 
2. De Np's Quiero ver cantidad y diversidad de reclutamientos, a cada familia metabolica
   cuantos tiene y de donde provienen 
   Correlacion total np's recruitment vs total antismash NP's
2. De antismash 
   Familia taxonomica vs clusters
   cluster interesante por familia taxonomica
   Voy a usar Actinos Taxa tambien para ver esos reclutamientos de donde provienen
3. De HeatPlot quiero ordenarlo por filogenia 
Heat plot Familia taxonomica, 
boxplot reclutamientos para cada familia metabolica
y hacer heatplots separados por ruta metabolica
4. Inlcuir algunos arboles
-->

<!--   Correlacion total np's recruitment vs total antismash NP's
-->
## Central pathway expansions  
Heat plot of central pathways expansions, Needs to be phylogenetically sorted.  
```{r HeatPlotsActino, warnings = FALSE, message = FALSE, echo=FALSE,CACHE=TRUE}
### ME falta ordernarlos filogeneticamente
ActinosHeatPlot$RastId <- with(ActinosHeatPlot,reorder(ActinosHeatPlot$RastId, ActinosHeatPlot$RastId))

ActinosHeatPlot.m <- melt(ActinosHeatPlot)
ActinosHeatPlot.m<- ddply(ActinosHeatPlot.m, .(variable), transform,rescale=scale(value))  ## rescale escala toda la matriz, scale por columnas
HP_TaxaActino<-merge(ActinosHeatPlot,ActinosTaxa,by.x = "RastId",by.y = "RastId")

HP_TaxaActino<-HP_TaxaActino[order(HP_TaxaActino$Order,HP_TaxaActino$Family),]
HP_TaxaActino.m<- melt(HP_TaxaActino)
HP_TaxaActino.m<- ddply(HP_TaxaActino.m, .(variable), transform,rescale=scale(value))  ## rescale escala toda la matriz, scale por columnas

heatplotActinos<-ggplot(HP_TaxaActino.m, aes(HP_TaxaActino.m$variable, HP_TaxaActino.m$RastId))+ geom_tile(aes(fill = HP_TaxaActino.m$rescale), colour = "white") + scale_fill_gradientn(colours = hm.palette(100),na.value = "gray")+ theme(legend.text=element_text(size=6),text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5,size=2) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )

ggsave("chapter4/HeatPlotActinos.pdf", plot = heatplotActinos,height = 12, width = 6)

```

```{r HeatPlots_Actino_label, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter4/HeatPlotActinos.pdf", caption = "Actinobacterial Heatplot",scale=.6,label = "ActinoPlot", type = "figure")
```
#Here is a reference to the HeatPlot: `r ref("ActinoPlot")`. 
\clearpage 

```{r Heaplot_summary_Actino,echo=FALSE}
#summary(ActinosHeatPlot.m)
#table(ActinosHeatPlot.m$scale >=0,Deltas$DeltaPRO>=0)

```

<!---------------END EvoMining heatmap----------------------------------------------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->


<!--- Expansions Box plot ------------------------------------------------------------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->
### Expansions BoxPlot by metabolic family
```{r boxplot_ActinoHP,warnings = FALSE, message = FALSE,echo=FALSE}
expansion_plotActino<-ggplot(ActinosHeatPlot.m, aes(x=ActinosHeatPlot.m$variable, y=ActinosHeatPlot.m$rescale)) + geom_boxplot()+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6), legend.position = "bottom")

ggsave("chapter4/expansion_plotActinos.pdf", plot = expansion_plotActino,height = 6, width = 7)
```
```{r boxplot_ActinoHP_Actino_label, results = "asis",message=FALSE}
label(path = "chapter4/expansion_plotActinos.pdf", caption = "Expansions Boxplot",label = "Actino_expansion_boxplot", type = "figure")
```
Here is a reference to the expansion boxplot: `r ref("Actino_expansion_boxplot")`.   
\clearpage 

<!-----------------------------END expansion boxplot ---------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->
## Genome Size correlations
### Correlation between genome size and AntiSMASH products  
```{r ANTISMASH_GenomeSize_Correlation_by_ORG_Actinos2,echo=FALSE}
EXP_SMASH_Actino <- cbind(ActinosSMASH, TOTAL = rowSums(ActinosSMASH))

EXP_SMASH_Actino<-merge(ActinosHeatPlot,ActinosSMASH,by.x="RastId",by.y="RastId")
EXP_SMASH_Actino <- cbind(EXP_SMASH_Actino, TOTAL = rowSums(EXP_SMASH_Actino))

EXP_SMASH2_Actino<-merge(ActinosTaxa, EXP_SMASH_Actino, by.x="RastId",by.y="RastId") 
#qplot(Total.x, Total.y, data = EXP_SMASH, colour=Order)+theme_bw()+
sn1Actino<-qplot(Size, TOTAL.y, data = EXP_SMASH2_Actino,colour=Order)+theme_bw()+ theme(legend.position	="bottom",plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))

ggsave("chapter4/SMASHvsSizebyOrderActinos.pdf", plot = sn1Actino,height = 8, width = 6)

sn2Actino<-qplot(Size, TOTAL.y, data = EXP_SMASH2_Actino,colour=Order)+theme_bw()+facet_grid(EXP_SMASH2_Actino$Order~.,scales="free")+ theme(legend.position	="bottom",plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
ggsave("chapter4/SMASHvsSizeGridOrderActinos.pdf", plot = sn2,height = 8, width = 6)

## Looking for color modifications
#sn + scale_color_brewer(palette="Dark2(100)")
#n + scale_color_brewer(palette=hm.palette(10))
## sn+ scale_color_hue(l=40, c=35)
```

