---
header-includes:
- \usepackage{graphicx,latexsym}
- \usepackage{amssymb,amsthm,amsmath}
- \usepackage{longtable,booktabs,setspace}
output: pdf_document
---

<!--- Cyanobacteria-->

```{r include_packages_5, include = FALSE,echo=FALSE}
# This chunk ensures that the reedtemplates package is
# installed and loaded. This reedtemplates package includes
# the template files for the thesis and also two functions
# used for labeling and referencing
if(!require(devtools))
  install.packages("devtools", repos = "http://cran.rstudio.com")
if(!require(dplyr))
    install.packages("dplyr", repos = "http://cran.rstudio.com")
if(!require(ggplot2))
    install.packages("ggplot2", repos = "http://cran.rstudio.com")
if(!require(reedtemplates)){
  library(devtools)
  devtools::install_github("ismayc/reedtemplates")
  }
library(reedtemplates)
flights <- read.csv("data/flights.csv")
```

<!-- Loading data
-->

```{r loading_tables_5,message=FALSE,echo=FALSE}
library(ggplot2)
library( reshape )
library(plyr )
library(RColorBrewer)
library(scales)
library(plyr)
library(RColorBrewer)
hm.palette <- colorRampPalette(rev(brewer.pal(11, 'Spectral')), space='Lab')  

#setwd("/home/nelly/GIT/phd/Thesis/tesis/chap5_files")
CyanosCentral <- read.table("chapter5/CyanosCentral", header=TRUE, sep="\t")
CyanosHeatPlot <- read.table("chapter5/CyanosHeatPlot", header=TRUE, sep="\t")
CyanosNp <- read.table("chapter5/CyanosNp", header=TRUE, sep="\t")
CyanosSMASH <- read.table("chapter5/CyanosSMASH", header=TRUE, sep="\t")
CyanosTaxa <- read.table("chapter5/CyanosTaxa", header=TRUE, sep="\t")
```


```{r CyanoSumary,echo=FALSE}
#summary(CyanosTaxa)
#table(CyanosTaxa$Order)
```


# Cyanobacteria EvoMining Results {#ref_labels}
Cyanobacteria is a photosynthetic phylum that inhabits a broad range of habitats. The broad adaptive potential is on part driven by gene-family enlargment [@larsson_genome_2011] by the analysis of 58 Cyanobacterial genomes concludes ancestor of cyanobacteria had a genome size of approx. 4.5 Mbp.
Cyanobacteria produces natural products as pigments and toxins [@whitton_ecology_2012]   Example of a PriA cluster toxins[@moustafa_origin_2009]

Fossil record situates Cyanobacteria [@whitton_ecology_2012]
Molecular record and metabolic propoerties at [@battistuzzi_genomic_2004]
  
Tables
-------------------------------------------------------------------
  Factors                    Correlation between Parents & Child     
------------------------- ----------------------------------------- 
  GenomeDB                                  1245                         
  Families                                  65                          
------------------------- -----------------------------------------
Table: Families on Cyanobacteria \label{tab:inher}

\clearpage
<!-- clearpage ends the page, and also dumps out all floats.Floats are things like tables and figures. 
-->
<!--
Plan general 
2. De Np's Quiero ver cantidad y diversidad de reclutamientos, a cada familia metabolica
   cuantos tiene y de donde provienen 
   Correlacion total np's recruitment vs total antismash NP's
2. De antismash 
   Familia taxonomica vs clusters
   cluster interesante por familia taxonomica
   Voy a usar Cyanos Taxa tambien para ver esos reclutamientos de donde provienen
3. De HeatPlot quiero ordenarlo por filogenia 
Heat plot Familia taxonomica, 
boxplot reclutamientos para cada familia metabolica
y hacer heatplots separados por ruta metabolica
4. Inlcuir algunos arboles
-->

<!--  EvoMining Expansions
-->


<!--- Expansions Box plot ------------------------------------------------------------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->
### Expansions BoxPlot by metabolic family
```{r boxplot_CyanoHP,warnings = FALSE, message = FALSE,echo=FALSE}
#CyanosHeatPlot$RastId <- with(CyanosHeatPlot,reorder(CyanosHeatPlot$RastId, CyanosHeatPlot$RastId))
CyanosTotalBP.m<-merge(CyanosHeatPlot,CyanosTaxa,by.x="RastId",by.y="RastId") ## works as expected
CyanosHeatPlotBP.m <- melt(CyanosTotalBP.m,id =c("RastId","Name","Phylum","Class","Order","Family","RastNo","Size","Contigs"))
CyanosHeatPlotBP.m<-subset(CyanosHeatPlotBP.m,variable!="TOTAL") ## works as expected

expansion_plotCyano<-ggplot(CyanosHeatPlotBP.m, aes(x=CyanosHeatPlotBP.m$variable, y=CyanosHeatPlotBP.m$value),color="Family") + labs(x = "Metabolic Families", y = "Copies on Cyanobacteria Genomes",text = element_text(size=12)) + geom_boxplot()+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6), legend.position = "bottom")

ggsave("chapter5/expansion_plotCyanos.pdf", plot = expansion_plotCyano,height = 8, width = 7)
```
```{r boxplot_CyanoHP_Cyano_label, results = "asis",message=FALSE}
label(path = "chapter5/expansion_plotCyanos.pdf", caption = "Expansions Boxplot",label = "Cyano_expansion_boxplot", type = "figure")
```
Here is a reference to the expansion boxplot: `r ref("Cyano_expansion_boxplot")`.   
\clearpage 

<!----------------------Expansions Box plot END-------------------------------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->






## Central pathway expansions  
Heat plot of central pathways expansions, Needs to be phylogenetically sorted.  
```{r HeatPlots, warnings = FALSE, message = FALSE, echo=FALSE,CACHE=TRUE}
### ME falta ordernarlos filogeneticamente
CyanosHeatPlot$RastId <- with(CyanosHeatPlot, reorder(CyanosHeatPlot$RastId, CyanosHeatPlot$TOTAL))
CyanosHeatPlot.m <- melt(CyanosHeatPlot)
CyanosHeatPlot.m<- ddply(CyanosHeatPlot.m, .(variable), transform,rescale=scale(value))  ## rescale escala toda la matriz, scale por columnas
HP_Taxa<-merge(CyanosHeatPlot,CyanosTaxa,by.x = "RastId",by.y = "RastId")
HP_Taxa2<-subset(HP_Taxa,Order='Chroococcales')

HP_Taxa<-HP_Taxa[order(HP_Taxa$Order,HP_Taxa$Family),]
HP_Taxa.m<- melt(HP_Taxa)
HP_Taxa.m<- ddply(HP_Taxa.m, .(variable), transform,rescale=scale(value))  ## rescale escala toda la matriz, scale por columnas

heatplot <- ggplot(HP_Taxa.m, aes(HP_Taxa.m$variable, HP_Taxa.m$RastId))+ geom_tile(aes(fill = HP_Taxa.m$rescale), colour = "white") + scale_fill_gradientn(colours = hm.palette(100),na.value = "gray")+ theme(text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )


#heatplot <- ggplot(CyanosHeatPlot.m, aes(CyanosHeatPlot.m$variable, CyanosHeatPlot.m$RastId))+ geom_tile(aes(fill = CyanosHeatPlot.m$rescale), colour = "white") + scale_fill_gradient(low = "white", high = "steelblue")+ theme(text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )

## este funciona pero lo esty tratando cambiar a ordenado
###heatplot <- ggplot(CyanosHeatPlot.m, aes(CyanosHeatPlot.m$variable, CyanosHeatPlot.m$RastId))+ geom_tile(aes(fill = CyanosHeatPlot.m$rescale), colour = "white") + scale_fill_gradientn(colours = hm.palette(100),na.value = "gray")+ theme(text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )


#+ scale_fill_gradientn(colours = hm.palette(100),na.value = "gray")
#+geom_text(aes(label=CyanosHeatPlot.m$rescale)) 
ggsave("chapter5/HeatPlot.pdf", plot = heatplot,height = 12, width = 6)

```

```{r HeatPlots_label, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter5/HeatPlot.pdf", caption = "Cyanobacterial Heatplot",scale=.6,label = "CyanoPlot", type = "figure")
```
Here is a reference to the HeatPlot: `r ref("CyanoPlot")`. 
\clearpage 

```{r Heaplot_summary,echo=FALSE}
#summary(CyanosHeatPlot.m)
#table(CyanosHeatPlot.m$scale >=0,Deltas$DeltaPRO>=0)

```

<!---------------END EvoMining heatmap----------------------------------------------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->







## Genome Size correlations
### Correlation between genome size and AntiSMASH products  
```{r ANTISMASH_GenomeSize_Correlation_by_ORG_Cyanos2,echo=FALSE}
EXP_SMASH<-merge(CyanosHeatPlot,CyanosSMASH,by.x="RastId",by.y="RastId")
EXP_SMASH2<-merge(CyanosTaxa, EXP_SMASH, by.x="RastId",by.y="RastId") 
#qplot(Total.x, Total.y, data = EXP_SMASH, colour=Order)+theme_bw()+
sn1<-qplot(Size, TOTAL.y, data = EXP_SMASH2,colour=Order)+theme_bw()+ theme(legend.position	="bottom",plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
ggsave("chapter5/SMASHvsSizebyOrder.pdf", plot = sn1,height = 8, width = 6)

sn2<-qplot(Size, TOTAL.y, data = EXP_SMASH2,colour=Order)+theme_bw()+facet_grid(EXP_SMASH2$Order~.,scales="free")+ theme(legend.position	="bottom",plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
ggsave("chapter5/SMASHvsSizeGridOrder.pdf", plot = sn2,height = 8, width = 6)

## Looking for color modifications
#sn + scale_color_brewer(palette="Dark2(100)")
#n + scale_color_brewer(palette=hm.palette(10))
## sn+ scale_color_hue(l=40, c=35)
```

Genome size vs Total antismash cluster coloured by order  
```{r SMASHvsSizebyOrder_label, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter5/SMASHvsSizebyOrder.pdf", caption = "Correlation between genome size and antismash Natural products detection colored by Order",scale=.6,label = "SMASHvsSizebyOrder", type = "figure")
```
Here is a reference to Genome size vs Total antismash cluster: `r ref("SMASHvsSizebyOrder")`. 
\clearpage

Genome size vs Total antismash cluster detected splitted by order  
```{r SMASHvsSizeGridOrder_label, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter5/SMASHvsSizeGridOrder.pdf", caption = "Correlation between genome size and antismash Natural products detection grided by Order",scale=.6,label = "SMASHvsSizeGridOrder", type = "figure")
```

Here is a reference to Correlation between genome size and antismash Natural products detection grided by Order plot: `r ref("SMASHvsSizeGridOrder")`. 
\clearpage 

<!--------------------------------END size vs AntiSMASH correlation--------------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->





<!-------------------------------- size vs Expansions correlation--------------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->
### Correlation between genome size and Central pathway expansions  
```{r ANTISMASH_GenomeSize_Correlation_by_ORG_Cyanos,echo=FALSE}
#CyanosHeatPlot$RastId <- with(CyanosHeatPlot, reorder(CyanosHeatPlot$RastId, CyanosHeatPlot$TOTAL))
#CyanosHeatPlot.m <- melt(CyanosHeatPlot)

EXP_TAXA<-merge(CyanosHeatPlot,CyanosTaxa,by.x="RastId",by.y="RastId")

sn4<-qplot(Size, TOTAL, data = EXP_TAXA,colour=Order)+theme_bw()+ theme(legend.position	="bottom",plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
ggsave("chapter5/SizevsExpansionsbyOrder.pdf", plot = sn4,height = 6, width = 6)

sn5<-qplot(Size, TOTAL, data = EXP_TAXA,colour=Order)+theme_bw()+facet_grid(EXP_TAXA$Order~.,scales="free")+ theme(legend.position	="bottom",plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
ggsave("chapter5/SizevsExpansionsGridbyOrder.pdf", plot = sn5,height = 6, width = 6)

## Looking for color modifications
#sn + scale_color_brewer(palette="Dark2(100)")
#n + scale_color_brewer(palette=hm.palette(10))
## sn+ scale_color_hue(l=40, c=35)
```
Genome size vs Total central pathway expansion coloured by order  
```{r SizevsExpansionsbyOrder_label2, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter5/SizevsExpansionsbyOrder.pdf", caption = "Correlation between genome size and central pathway expansions ",label = "SizevsExpansionsbyOrder", type = "figure")
```
Here is a reference to the size vs Total central pathway expansion plot: `r ref("SizevsExpansionsbyOrder")`. 
\clearpage 


Genome size vs Total central pathway expansion grided by order  
```{r SizevsExpansionsGridbyOrder_label2, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter5/SizevsExpansionsGridbyOrder.pdf", caption = "Correlation between genome size and central pathway expansions grided by order",label = "SizevsExpansionsGridbyOrder", type = "figure")
```
Here is a reference to the Genome size vs Total central pathway expansion grided by order plot: `r ref("SizevsExpansionsGridbyOrder")`. 
\clearpage 

<!--------------------------------------------------------------------------------------------------------------------->



Correlation between genome size and each of the central pathway families.
Data are coloured by metabolic family instead of coloured by taxonomical order. This treatment allows to answer how differente metabolic families grows when genome size grow.  
Also I want to add form given by taxonomical order.

```{r ANTISMASH_GenomeSize_Correlation_by_ORG_Cyanos3,message=FALSE, echo=FALSE}
CyanosHeatPlot$RastId <- with(CyanosHeatPlot, reorder(CyanosHeatPlot$RastId, CyanosHeatPlot$TOTAL))
CyanosHeatPlot.m <- melt(CyanosHeatPlot)
EXP_TAXA<-merge(CyanosHeatPlot.m,CyanosTaxa,by.x="RastId",by.y="RastId")

sn6<-qplot(Size, value, data = EXP_TAXA,colour=variable,shape=Order)+theme_bw()+ theme(legend.position	="none",legend.text=element_text(size=6),plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
ggsave("chapter5/SizevsExpansionsbyMetabolicFamily.pdf", plot = sn6,height = 6, width = 8)

sn7<-qplot(Size, value, data = EXP_TAXA,colour=variable)+theme_bw()+facet_grid(EXP_TAXA$variable~.,scales="free")+ theme(legend.position	="none",plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
ggsave("chapter5/ExpansionsbyMetabolicFamilyGrid.pdf", plot = sn7,height = 6, width = 12)

## Looking for color modifications
#sn + scale_color_brewer(palette="Dark2(100)")
#n + scale_color_brewer(palette=hm.palette(10))
## sn+ scale_color_hue(l=40, c=35)
```

Genome size vs Total central pathway expansion coloured by metabolic Family 
```{r SizevsExpansionsbyMetabolicFamily_label2, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter5/SizevsExpansionsbyMetabolicFamily.pdf", caption = "Correlation between Genome size vs Total central pathway expansion coloured by metabolic Family",scale=.6,label = "SizevsExpansionsbyMetabolicFamily", type = "figure")
```
Here is a reference to the Genome size vs Total central pathway expansion coloured by metabolic Family plot: `r ref("SizevsExpansionsbyMetabolicFamily")`. 
\clearpage 

Future Work: Genome size vs Total central pathway expansion grided by metabolic Family 
For clarity I need to also grid and group by Metabolic Pathway
```{r ExpansionsbyMetabolicFamilyGrid_label2, results = "asis",message=FALSE,echo=FALSE}
#label(path = "chapter5/ExpansionsbyMetabolicFamilyGrid.pdf", caption = "Correlation between Genome size vs Total central pathway expansion grided by metabolic Family ",label = "ExpansionsbyMetabolicFamilyGrid", type = "figure")
```
Here is a reference to Genome size vs Total central pathway expansion grided by metabolic Family  plot: `r ref("ExpansionsbyMetabolicFamilyGrid")`. 
\clearpage 


<!----------------------------END - size vs Expansions correlation--------------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->





<!----------------------------Natural Products vs Expansions correlation--------------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->

## Natural products  

### Natural products recruitments from EvoMining heatplot
We can see natural products recruitment after central pathways expansions colored by their kingdom.  
Natural products recruited by metabolic family, colored by phylogenetic origin.
```{r NP_recruitment_taxa_cyano,echo=FALSE,message=FALSE,warning=FALSE}
RecbyKing<-qplot(x=Central, data=CyanosNp, geom="bar", fill=kingdom, position="stack")+theme_bw()+ theme(legend.position	="bottom",text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )
ggsave("chapter5/RecruitmentsbyKingdom.pdf", plot = RecbyKing,height = 6, width = 12)

Recbytaxa<-qplot(x=Central, data=CyanosNp, geom="bar", fill=taxa, position="stack")+theme_bw()+ theme(legend.position	="bottom",text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )
ggsave("chapter5/RecruitmentsbyTaxa.pdf", plot = Recbytaxa,height = 6, width = 12)

```


Recruitments after central pathways expansions coloured by Kingdom
```{r RecruitmentsbyKingdom_label2, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter5/RecruitmentsbyKingdom.pdf", caption = "Recruitmens on central families coloured by kingdom",scale=.6,label = "RecruitmentsbyKingdom", type = "figure")
```
Here is a reference to Recruitments after central pathways expansions colourd by Kingdom plot: `r ref("RecruitmentsbyKingdom")`. 

\clearpage 
Recruitments after central pathways expansions colourd by taxonomy
```{r RecruitmentsbyTaxa_label2, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter5/RecruitmentsbyTaxa.pdf", caption = "Recruitmens on central families coloured by taxonomy",scale=.5,label = "RecruitmentsbyTaxa", type = "figure")
```
Here is a reference to Recruitments after central pathways expansions colourd by taxa plot: `r ref("RecruitmentsbyTaxa")`. 
\clearpage 






## Cyanobacterias AntiSMASH 
```{r CyanoDiversity,echo=FALSE,message=FALSE,warning=FALSE}
## merge tables
m1 <- merge(CyanosTaxa, CyanosSMASH, by.x = "RastId", by.y = "RastId")
m1.m <- melt(m1, id = c("RastId", "Name","Phylum","Class","Order","Family","Contigs","Size","RastNo"))
m2.m<-subset(m1.m,value>0)

Diversity<-qplot(x=Order, data=m1.m, geom="bar", fill=Family, position="stack")+theme_bw()+ theme(text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )
#table(m1.m$Order>0,m1.m$Family>0)
#qplot(x=clarity, data=diamonds, geom="bar", fill=cut, position="stack")
ggsave("chapter5/Diversity.pdf", plot =Diversity,height = 6, width = 14)

Smash<-qplot(x=Order, data=m2.m, geom="bar", fill=variable, position="stack",colours = hm.palette(100000))+theme_bw()+ theme(legend.position="none",text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )
#table(m1.m$Order>0,m1.m$Family>0)
ggsave("chapter5/Smash.pdf", plot = Smash,height = 6, width = 14)
```
 
Taxonomical diversity on Cyanobacteria Data
```{r Diversity_label2, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter5/Diversity.pdf", caption = "Diversity",scale=.6,label = "Diversity", type = "figure")
```
Here is a reference to Recruitments after central pathways expansions colourd by taxa plot: `r ref("Diversity")`. 
\clearpage

Smash diversity
```{r Smash_label2, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter5/Smash.pdf", caption = "Smash",scale=.5,label = "Smash Taxonomical Diversity", type = "figure")
```
Here is a reference to Recruitments after central pathways expansions colourd by taxa plot: `r ref("Smash")`. 
\clearpage





### AntisSMASH vs Central Expansions  
Is it a correlation between pangenome grow and central pathways expansions?  
```{r ANTISMASH_Expansion_Correlation_by_ORG_Cyanos,message=FALSE, echo =FALSE}
EXP_SMASH<-merge(CyanosHeatPlot,CyanosSMASH,by.x="RastId",by.y="RastId")
EXP_SMASH2<-merge(CyanosTaxa, EXP_SMASH, by.x="RastId",by.y="RastId") 
#qplot(Total.x, Total.y, data = EXP_SMASH, colour=Order)+theme_bw()+
sn8<-qplot(TOTAL.x, TOTAL.y, data = EXP_SMASH2,colour=Order)+theme_bw()+ theme(legend.position	="bottom",legend.text=element_text(size=10),plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 10),axis.text.y=element_text(size = 7))
ggsave("chapter5/SMASHvsExpansionsbyOrder.pdf", plot = sn8,height = 12, width = 6)

sn9<-qplot(TOTAL.x, TOTAL.y, data = EXP_SMASH2,colour=Order)+theme_bw()+facet_grid(EXP_SMASH2$Order~.,scales="free")+ theme(legend.position	="bottom",legend.text=element_text(size=10),plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
ggsave("chapter5/SMASHvsExpansionsbyOrderGRID.pdf", plot = sn9,height = 10, width = 6)

## Looking for color modifications
#sn + scale_color_brewer(palette="Dark2(100)")
#n + scale_color_brewer(palette=hm.palette(10))
## sn+ scale_color_hue(l=40, c=35)
```
Total central pathway expansions by genome vs Total antismash cluster detected coloured by order  
```{r SMASHvsExpansionsbyOrder_label, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter5/SMASHvsExpansionsbyOrder.pdf", caption = "Correlation between central pathway axpnasions and antismash Natural products detection",scale=.5,label = "SMASHvsExpansionsbyOrder", type = "figure")
```
Here is a reference to the expansions vs antismash NP's clusters plot: `r ref("SMASHvsExpansionsbyOrder")`. 
\clearpage 

Total central pathway expansions by genome vs Total antismash cluster detected splitted by order  
```{r SMASHvsExpansionsbyOrderGRID_label, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter5/SMASHvsExpansionsbyOrderGRID.pdf", caption = "Correlation between central pathway axpnasions and antismash Natural products detection",scale=.5,label = "SMASHvsExpansionsbyOrderGRID", type = "figure")
```
Here is a reference to the expansions vs antismash NP's clusters splitted by order plot `r ref("SSMASHvsExpansionsbyOrderGRID")`. 
\clearpage 

<!------------------------------------------------------------------------------------------------------------->




AntisMAsh vs Expansions by taxonomic Family
```{r ANTISMASH_Expansion_Correlation_by_Family_Cyanos,echo=FALSE,warning=FALSE}
EXP_SMASH<-merge(CyanosHeatPlot,CyanosSMASH,by.x="RastId",by.y="RastId")
EXP_SMASH2<-merge(CyanosTaxa, EXP_SMASH, by.x="RastId",by.y="RastId") 

npf<-qplot(TOTAL.x, TOTAL.y, data = EXP_SMASH2,colour=Family)+theme_bw()+facet_grid(EXP_SMASH2$Order~.,scales="free")+
        theme( plot.title = element_text(size = 14, face = "bold"),
              text = element_text(size = 12),
              axis.title = element_text(face="bold"),
              axis.text.x=element_text(size = 12),
              axis.text.y=element_text(size = 7))
ggsave("chapter5/npf.pdf", plot = npf,height = 12, width = 6)

#facet_grid(EXP_SMASH$Order~.,scales="free")+
```

Natural products colured by family
```{r npf_label, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter5/npf.pdf", caption = "Natural products by family",scale=.6,label = "npf", type = "figure")
```
Here is a reference to the Natural products colured by family
 plot `r ref("npf")`. 
\clearpage 

<!------------------------------------------------------------------------------------------------------------->

##Selected trees from EvoMining
Phosphoribosyl_isomerase_3 family  
Figure from EvoMining

```{r Phosphoribosyl_isomerase_evo_tree, results = "asis",message=FALSE, echo=FALSE}

label(path = "chapter5/tree64.png", caption = "Phosphoribosyl isomerase EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "Phosphoribosyl_isomerase_evo_tree", type = "figure")

label(path = "chapter5/tree1.png", caption = "Phosphoglycerate dehydrogenase EvoMiningtree",scale = .25,
      angle = 180,
      label = "Phosphoglycerate_dehydrogenase_evo_tree", type = "figure")

label(path = "chapter5/tree2.png", caption = "Phosphoserine aminotransferase EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "Phosphoserine_aminotransferase_evo_tree", type = "figure")

label(path = "chapter5/tree10.png", caption = "Triosephosphate isomerase EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "Triosephosphate_isomerase_evo_tree", type = "figure")

label(path = "chapter5/tree11.png", caption = "glyceraldehyde3phosphate dehydrogenase EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "glyceraldehyde3phosphate_dehydrogenase_evo_tree", type = "figure")

label(path = "chapter5/tree12.png", caption = "phosphoglycerate kinase EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "phosphoglycerate_kinase_evo_tree", type = "figure")

label(path = "chapter5/tree13.png", caption = "phosphoglycerate mutaseEvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "phosphoglycerate_mutase_evo_tree", type = "figure")

label(path = "chapter5/tree14.png", caption = "enolase EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "enolase_evo_tree", type = "figure")

label(path = "chapter5/tree15.png", caption = "Pyruvate kinase EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "Pyruvate_kinase_evo_tree", type = "figure")

label(path = "chapter5/tree16.png", caption = "Aspartate transaminase EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "Aspartate_transaminase_evo_tree", type = "figure")

label(path = "chapter5/tree17.png", caption = "Asparagine synthase EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "Asparagine_synthase_evo_tree", type = "figure")

label(path = "chapter5/tree18.png", caption = "Aspartate kinase EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "Aspartate_kinase_evo_tree", type = "figure")

label(path = "chapter5/tree19.png", caption = "Aspartate semialdehyde dehydrogenase EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "Aspartate_semialdehyde_dehydrogenase_evo_tree", type = "figure")


label(path = "chapter5/tree20.png", caption = "Homoserine dehydrogenase EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "Homoserine_dehydrogenase_evo_tree", type = "figure")
#label(path = "chapter5/tree64.png", caption = "Tree 64", 

```
\clearpage 
