---
header-includes:
- \usepackage{graphicx,latexsym}
- \usepackage{amssymb,amsthm,amsmath}
- \usepackage{longtable,booktabs,setspace}
output: pdf_document
---
<!--- Archaeas-->

<!--
You can delete the header-includes (lines 3-6 above) if you like and also the chunk below since it is loaded in the skeleton.Rmd file.  They are included so that chap3.Rmd will compile by itself when you hit Knit PDF.
-->

```{r include_packages_2, include = FALSE}
# This chunk ensures that the reedtemplates package is
# installed and loaded. This reedtemplates package includes
# the template files for the thesis and also two functions
# used for labeling and referencing
if(!require(devtools))
  install.packages("devtools", repos = "http://cran.rstudio.com")
if(!require(plyr))
    install.packages("plyr", repos = "http://cran.rstudio.com") ## this shoul always go before dplyr
if(!require(dplyr))
    install.packages("dplyr", repos = "http://cran.rstudio.com")
if(!require(ggplot2))
    install.packages("ggplot2", repos = "http://cran.rstudio.com")
if(!require(reedtemplates)){
  library(devtools)
  devtools::install_github("ismayc/reedtemplates")
  }
library(reedtemplates)
```


```{r loading_tables_Archaea,echo=FALSE,message=FALSE}
library( reshape )
library(RColorBrewer)
library(scales)
library(RColorBrewer)
library(ggplot2)
tropical=c('darkorange','dodgerblue','hotpink')
palette(tropical)

hm.palette <- colorRampPalette(rev(brewer.pal(11, 'Spectral')), space='Lab')  

ArchaeasCentral <- read.table("chapter3/ArchaeasCentral", header=TRUE, sep="\t")
ArchaeasHeatPlot <- read.table("chapter3/ArchaeasHeatPlot", header=TRUE, sep="\t")
ArchaeasNp <- read.table("chapter3/ArchaeasNp", header=TRUE, sep="\t")
ArchaeasSMASH <- read.table("chapter3/ArchaeasSMASH", header=TRUE, sep="\t")
ArchaeasTaxa <- read.table("chapter3/ArchaeasTaxa", header=TRUE, sep="\t")
```

```{r creating_practiceSet}
ArchaeasHeatPlot <- read.table("chapter3/ArchaeasHeatPlot", header=TRUE, sep="\t")
ArchaeasTaxa <- read.table("chapter3/ArchaeasTaxa", header=TRUE, sep="\t")

## Adding order variable 
ArchaeasHeatPlot$order<-c(1:nrow(ArchaeasHeatPlot))

#sorting RastId it accordig to order variable
ArchaeasHeatPlot$RastId <- with(ArchaeasHeatPlot,reorder(ArchaeasHeatPlot$RastId, ArchaeasHeatPlot$order))

# Merging heatplot and taxonomy table into one table
HP_Archaeas_Taxa<-merge(ArchaeasHeatPlot,ArchaeasTaxa,by.x = "RastId",by.y = "RastId")
HP_Archaeas_Taxa.m<- melt(HP_Archaeas_Taxa)

HP_Archaeas_Taxa.m<- ddply(HP_Archaeas_Taxa.m, .(variable), transform,rescale=scale(value))  ## 

small<-HP_Archaeas_Taxa.m[which(HP_Archaeas_Taxa.m$RastId=='389007' | HP_Archaeas_Taxa.m$RastId=='390212' | HP_Archaeas_Taxa.m$RastId=='388991' | HP_Archaeas_Taxa.m$RastId=='388992' | HP_Archaeas_Taxa.m$RastId=='388199'| HP_Archaeas_Taxa.m$RastId=='388200'|  HP_Archaeas_Taxa.m$RastId=='388201'),]

#Este Escogiendo que variables quiero
small<-small[small$variable %in% c("Phosphoglycerate_dehydrogenase", "Isopropyl_malate_synthase","Alanine_dehydrogenase","Asparagine_synthase"), ] 
## use http://colorbrewer2.org/ to find optimal divergent color palette (or set own)
 
small.m<- ddply(small, .(variable),summarize, mean = round(mean(value), 2),sd = round(sd(value), 2),expansion=mean+sd, reduction=mean-sd)  ## 
rownames(small.m)<-small.m$variable
```



```{r coloring_test_set}
color_exp<-function(x){
  # Pendiente pasarle un dataframe en lugar de tener fijo small.m como variable global
  result=0
  expansion<-NULL
  reduction<-NULL
  local_value=x[1,"value"]
  met_family<-x[1,"variable"]
  Rast=x[1,1]
  #print ("rast",Rast)
  # print(paste(x,"family",met_family,"value",local_value,"Rast",Rast))
  expansion<-small.m$expansion [which(small.m$variable==met_family)] 
  reduction<-small.m$reduction [which(small.m$variable==met_family)] 
  if (local_value>=expansion){result= 1}
  else if (local_value<=reduction){result= -1}
  return (result)
}

small.expansion<-adply(small,1,color_exp)

#small.expansion$V1
#color_palette <- colorRampPalette(c("#3794bf", "#FFFFFF","#df8640"))

ggplot(small.expansion, aes(small.expansion$variable, small.expansion$RastId,label=round(small.expansion$value)))+ geom_tile(aes(fill = small.expansion$V1), colour = "white") + scale_fill_gradientn(colours = hm.palette(100),na.value = "gray")+ theme(text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5))+geom_text(size=4,check_overlap = TRUE)

```


```{r learningHeatPlots}
########################################################
## Trying to sort the heatplot 
## Reading heatplot table and taxa information and saving it on data.frame data structure
ArchaeasHeatPlot <- read.table("chapter3/ArchaeasHeatPlot", header=TRUE, sep="\t")
ArchaeasTaxa <- read.table("chapter3/ArchaeasTaxa", header=TRUE, sep="\t")
hm.palette <- colorRampPalette(rev(brewer.pal(11, 'Spectral')), space='Lab')  
## Adding order variable 
ArchaeasHeatPlot$order<-c(1:nrow(ArchaeasHeatPlot))
#sorting RastId it accordig to order variable
ArchaeasHeatPlot$RastId <- with(ArchaeasHeatPlot,reorder(ArchaeasHeatPlot$RastId, ArchaeasHeatPlot$order))
# Merging heatplot and taxonomy table into one table
HP_Archaeas_Taxa<-merge(ArchaeasHeatPlot,ArchaeasTaxa,by.x = "RastId",by.y = "RastId")
# Melting information leting as variables just enzymatic families and copy number
HP_Archaeas_Taxa.m<- melt(HP_Archaeas_Taxa)

# Cleaning data
HP_Archaeas_Taxa.m<- ddply(HP_Archaeas_Taxa.m, .(variable), transform,value)  ## 
HP_Archaeas_Taxa.m<- ddply(HP_Archaeas_Taxa.m, .(variable), transform,rescale=scale(value))  ## 

HP_Archeas.calcs<- ddply(HP_Archaeas_Taxa.m, .(variable),summarize, mean = round(mean(value), 2),sd = round(sd(value), 2),expansion=mean+sd, reduction=mean-sd)  ## 

rownames(HP_Archeas.calcs)<-HP_Archeas.calcs$variable
#########################################################3
color_exp<-function(x){
  # Pendiente pasarle un dataframe en lugar de tener fijo small.m como variable global
  result=0
  expansion<-NULL
  reduction<-NULL
  local_value=x[1,"value"]
  met_family<-x[1,"variable"]
  Rast=x[1,1]
  #print ("rast",Rast)
  # print(paste(x,"family",met_family,"value",local_value,"Rast",Rast))
  expansion<-HP_Archeas.calcs$expansion [which(small.m$variable==met_family)] 
  reduction<-HP_Archeas.calcs$reduction [which(small.m$variable==met_family)] 
  if (local_value>=expansion){result= 1}
  else if (local_value<=reduction){result= -1}
  return (result)
}
small[ !small$variable %in% c("Contigs", "Size","TOTAL"), ] 
##Bueno aqui voy
##Heat.expansion<-adply(HP_Archaeas_Taxa.m[!HP_Archaeas_Taxa.m$variable %in% c("Contigs", "Size","TOTAL","order","RastNo"), ],1,color_exp)


#####################################333
#############################################


## geom tile with rescale rescala por column
        # Graph! with rescale
ggplot(small, aes(small$variable, small$RastId,label=round(small$rescale,digits=1)))+ geom_tile(aes(fill = small$rescale), colour = "white") + scale_fill_gradientn(colours = hm.palette(100),na.value = "gray")+ theme(text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5))+geom_text(size=4,check_overlap = TRUE)

## Graph scaled according to whole matrix 
ggplot(small, aes(small$variable, small$RastId,label=round(small$value,digits=1)))+ geom_tile(aes(fill = small$value), colour = "white") + scale_fill_gradientn(colours = hm.palette(100),na.value = "gray")+ theme(text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5))+geom_text(size=4,check_overlap = TRUE)

## Graph coloured by columns, showing true values, not scaled
ggplot(small, aes(small$variable, small$RastId,label=round(small$value,digits=1)))+ geom_tile(aes(fill = small$rescale), colour = "white") + scale_fill_gradientn(colours = hm.palette(100),na.value = "gray")+ theme(text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5))+geom_text(size=4,check_overlap = TRUE)

####################################################
### Color according to mean plus standar deviation
####

################# trying to make heatplot colors on dataframe

small.m<- ddply(small, .(variable),summarize, mean = round(mean(value), 2),sd = round(sd(value), 2),expansion=mean+sd, reduction=mean-sd)  ## 
rownames(small.m)<-small.m$variable

####################################################################
```



# Archaea EvoMining Results {#ref_labels}   
During the decade between 1970 and 1980, Archaea was recognized as new life domain, a kingdom different from Bacteria and Eucarya in an exciting first great application of 16S phylogeny[@woese_phylogenetic_1977,@woese_are_1981] . Main differences between this kingdoms are that Archaeal DNA is not arranged in a nucleus as in Eucarya and Archaeal celular walls are not composed from peptidoglycans as in Bacteria. Archaeal proteins may be higlhy valuable to biotechnology industry for their great stability due to extreme temperature, PH and salt content conditions on Archeal habitats. Despite no Archaeal Natural products biosynthetic gene clusters (BGC's) has been reported on MiBIG, Archaea do have BGC's, some of them seems to be acquired by horizontal gene transfer (HGT) like methano nrps {search reference}. Other Archeal natural products known are archaeosins, Diketopiperazines, Acyl Homoserine Lactones, Exopolysaccharides, Carotenoids, Biosurfactants, Phenazines and Organic Solutes but this knowledge is not comparable to Bacterial BGC's knowledge[@charlesworth_untapped_2015].  
  
Natural products biosynthetic gene clusters search is actually performed using either *high-confidence/low-novelty or low-confidence/high-novelty* bioinformatic approaches [@medema_computational_2015]. High confidence methods compares query sequences with previously known BGC's such as nrps or PKS, examples of this algorithms are antiSMASH and clusterfinder [@_antismash_????]. EvoMining searches on expansions from central metabolic pathways enzyme families, it has been classified as low confidence/high novelty method. EvoMining has proved useful on Actinobacteria phylum where its use lead to Arseno-compounds discovery [@cruz-morales_phylogenomic_2016]. Also on Actinobacteria antiSMASH analysis on 1245 genomes found 774 different classes of natural products, the same analysis on 876 Archaeal genomes, a full kingdom, identifies only 35 BGC's classes. So either Archaea does not have natural products BGC's or this are not yet known. Next paragraph deals with a possible approach about how natural products BGC's can be find.  
  
Archaea resembled Bacteria in that Archaea uses horizontal gene transfer as a genic interchange mecanism, Archaeal genomes contains operons [@howland_surprising_2000] and in general there is introns absence{Reference to Computational Methods for Understanding Bacterial and Archaeal Genomes}. Archaeas do have introns, but they are mainly located on genes that encodes ribosomal and transfer RNA [@howland_surprising_2000]. General lack of introns allows automatic genome annotation, operons gene organization permits functional inference to a certain degree and HGT contribute to expansions on Archaeal genomes. Some phylum on Archaea has an open pangenome, and as we will show on this chapter some Archaea has central pathway expansions. 
Enzyme families from central pathways expansions, open pangenome and operon organization made EvoMining succesful on Actinobacteria, this lead us to think that evoMining is suitable to analize Archaeal genomes, even more since EvoMining is a method oriented to use evolution and its not entirelyy based on previous knowledge of BGC's sequences if evolutionary logic behave on Archaea as on bacteria, new BGC's classes may be be found on Archaea.
  
EvoMining is a trade off between conserved known central metabolic function and enough expansions divergence on sequence and on clusters to divergence 

```{r ArchaeasSumary,echo=FALSE}
#summary(ArchaeasTaxa)
#table(ArchaeasTaxa$Order)
```

  
Tables
-------------------------------------------------------------------
  Factors                    Correlation between Parents & Child     
------------------------- ----------------------------------------- 
  GenomeDB                                  876                         
  Phylum                                    12   
  Order                                     23 
------------------------- -----------------------------------------
Table: Families on Archaeabacteria \label{tab:inher}

\clearpage

```{r ArchealResults, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter3/PosterArcheas.pdf", caption = "EvoMining Archaeas",scale=.9,label = "EvoMining Archaeas", type = "figure")
```

<!--  EvoMining Expansions
-->


<!--- Expansions Box plot ------------------------------------------------------------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->
First lets investigate if Archaea has expansions on families within central metabolic routes. Since main metabolic pathways are shared between Bacteria and Archaea makes sense to assemble Archeal EvoMining central database by using orthologous from Actinobacteria evoMining central pathways.

### Expansions BoxPlot by metabolic family
```{r boxplot_ArchaeaHP,warnings = FALSE, message = FALSE,echo=FALSE}
#ArchaeasHeatPlot$RastId <- with(ArchaeasHeatPlot,reorder(ArchaeasHeatPlot$RastId, ArchaeasHeatPlot$RastId))
ArchaeasTotalBP.m<-merge(ArchaeasHeatPlot,ArchaeasTaxa,by.x="RastId",by.y="RastId") ## works as expected
ArchaeasHeatPlotBP.m <- melt(ArchaeasTotalBP.m,id =c("RastId","Name","SuperPhylum","Phylum","Class","Order","Family","RastNo","Size","Contigs"))
ArchaeasHeatPlotBP.m<-subset(ArchaeasHeatPlotBP.m,variable!="TOTAL") ## works as expected

expansion_plotArchaea<-ggplot(ArchaeasHeatPlotBP.m, aes(x=ArchaeasHeatPlotBP.m$variable, y=ArchaeasHeatPlotBP.m$value),color="Family") + labs(x = "Metabolic Families", y = "Copies on Archaeabacteria Genomes",text = element_text(size=12)) + geom_boxplot()+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6), legend.position = "bottom")

ggsave("chapter3/expansion_plotArchaeas.pdf", plot = expansion_plotArchaea,height = 8, width = 7)
```

```{r boxplot_ArchaeaHP_Archaea_label, results = "asis",message=FALSE}
label(path = "chapter3/expansion_plotArchaeas.pdf", caption = "Expansions Boxplot",label = "Archaea_expansion_boxplot", type = "figure",scale=".7")
```
Here is a reference to the expansion boxplot: `r ref("Archaea_expansion_boxplot")`.   
\clearpage 


### Expansions BoxPlot by metabolic family by phylum
```{r ArcheaeBoxPlotByPhylum}
#+ geom_jitter()
#aes(fill = factor(vs))

ArchaeasTotalBP.m<-merge(ArchaeasHeatPlot,ArchaeasTaxa,by.x="RastId",by.y="RastId") ## works as expected
ArchaeasHeatPlotBP.m <- melt(ArchaeasTotalBP.m,id =c("RastId","Name","SuperPhylum","Phylum","Class","Order","Family","RastNo","Size","Contigs"))
ArchaeasHeatPlotBP.m<-subset(ArchaeasHeatPlotBP.m,variable!="TOTAL") ## works as expected
ArchaeasHeatPlotBP.m<-subset(ArchaeasHeatPlotBP.m,variable!="TOTAL") ## works as expected

## Each metabolic pathway se parte por phylum coloreado por order

#3PGA_AMINOACIDS
#Glycolysis
#OXALACETATE_AMINOACIDS
#R5P_AMINOACIDS
#TCA
#E4P_AMINO_ACIDS
#PYR_THR_AA

## Genome size
ggplot(ArchaeasHeatPlotBP.m, aes(x=ArchaeasHeatPlotBP.m$Phylum, y=ArchaeasHeatPlotBP.m$Size))+ geom_boxplot() +theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6), legend.position = "bottom")+ labs(x = "Copies on Archaeabacteria taxonomic groups", y = "Genome size",text = element_text(size=12))+theme_bw()
#+ geom_jitter(aes(color=ArchaeasHeatPlotBP.m$Phylum))


## Halobacteria
MetFam_BP.m=subset(ArchaeasHeatPlotBP.m,Phylum=="Euryarchaeota")
ggplot(MetFam_BP.m, aes(x=MetFam_BP.m$Order, y=MetFam_BP.m$Size))+ geom_boxplot() +theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 10), legend.position = "bottom")+ labs(x = "Halobacteria orders size taxonomic groups", y = "Genome size",text = element_text(size=12)) + geom_jitter(aes(color=Family))


#MetFam_BP.m=subset(ArchaeasHeatPlotBP.m,Family=="Methanosarcinaceae")
#ggplot(MetFam_BP.m, aes(x=MetFam_BP.m$Size, y=MetFam_BP.m$value))
#+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 10), legend.position = "bottom")+ labs(x = "Copies on Archaeabacteria taxonomic groups", y = "Genome size",text = element_text(size=12)) 


  #geom_jitter(aes(color=ArchaeasHeatPlotBP.m$Phylum))# + facet_grid(. ~ Phylum)+theme_bw()


## Metabolic Pathways
MetFam=subset(ArchaeasCentral,Pathway=="PPP")
MetFam_BP.m=ArchaeasHeatPlotBP.m[ArchaeasHeatPlotBP.m$variable %in% MetFam$Enzyme,]
ggplot(MetFam_BP.m, aes(x=MetFam_BP.m$variable, y=MetFam_BP.m$value, fill=Order))+ labs(x = "Metabolic PPP Families", y = "Copies on Archaeabacteria Genomes",text = element_text(size=12)) + geom_boxplot() + facet_grid(. ~ Phylum)+theme_bw() +theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6), legend.position = "bottom")
#+ geom_jitter(aes(color=MetFam_BP.m$Phylum))       

MetFam=subset(ArchaeasCentral,Pathway=="3PGA_AMINOACIDS")
MetFam_BP.m=ArchaeasHeatPlotBP.m[ArchaeasHeatPlotBP.m$variable %in% MetFam$Enzyme,]
ggplot(MetFam_BP.m, aes(x=MetFam_BP.m$variable, y=MetFam_BP.m$value, fill=Order))+ labs(x = "Metabolic PGA_AMINOACIDS Families", y = "Copies on Archaeabacteria Genomes",text = element_text(size=12)) + geom_boxplot() + facet_grid(. ~ Phylum)+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6), legend.position = "bottom")
#+ geom_jitter(aes(color=MetFam_BP.m$Phylum))

MetFam=subset(ArchaeasCentral,Pathway=="Glycolysis")
MetFam_BP.m=ArchaeasHeatPlotBP.m[ArchaeasHeatPlotBP.m$variable %in% MetFam$Enzyme,]
ggplot(MetFam_BP.m, aes(x=MetFam_BP.m$variable, y=MetFam_BP.m$value, fill=Order))+ labs(x = "Metabolic Glycolysis Families", y = "Copies on Archaeabacteria Genomes",text = element_text(size=12)) + geom_boxplot() + facet_grid(. ~ Phylum)+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6), legend.position = "bottom")
#+ geom_jitter(aes(color=MetFam_BP.m$Phylum))

MetFam=subset(ArchaeasCentral,Pathway=="OXALACETATE_AMINOACIDS")
MetFam_BP.m=ArchaeasHeatPlotBP.m[ArchaeasHeatPlotBP.m$variable %in% MetFam$Enzyme,]
ggplot(MetFam_BP.m, aes(x=MetFam_BP.m$variable, y=MetFam_BP.m$value, fill=Order))+ labs(x = "Metabolic OXALACETATE_AMINOACIDS Families", y = "Copies on Archaeabacteria Genomes",text = element_text(size=12)) + geom_boxplot() + facet_grid(. ~ Phylum)+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6), legend.position = "bottom")
#+ geom_jitter(aes(color=MetFam_BP.m$Phylum))

MetFam=subset(ArchaeasCentral,Pathway=="R5P_AMINOACIDS")
MetFam_BP.m=ArchaeasHeatPlotBP.m[ArchaeasHeatPlotBP.m$variable %in% MetFam$Enzyme,]
ggplot(MetFam_BP.m, aes(x=MetFam_BP.m$variable, y=MetFam_BP.m$value, fill=Order))+ labs(x = "Metabolic R5P_AMINOACIDS Families", y = "Copies on Archaeabacteria Genomes",text = element_text(size=12)) + geom_boxplot() + facet_grid(. ~ Phylum)+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6), legend.position = "bottom")
 #+ geom_jitter(aes(color=MetFam_BP.m$Phylum))

#
MetFam=subset(ArchaeasCentral,Pathway=="TCA")
MetFam_BP.m=ArchaeasHeatPlotBP.m[ArchaeasHeatPlotBP.m$variable %in% MetFam$Enzyme,]
ggplot(MetFam_BP.m, aes(x=MetFam_BP.m$variable, y=MetFam_BP.m$value, fill=Order))+ labs(x = "Metabolic TCA Families", y = "Copies on Archaeabacteria Genomes",text = element_text(size=12)) + geom_boxplot() + facet_grid(. ~ Phylum)+theme_bw() +theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6), legend.position = "bottom")
#+ geom_jitter(aes(color=MetFam_BP.m$Phylum))

MetFam=subset(ArchaeasCentral,Pathway=="E4P_AMINO_ACIDS")
MetFam_BP.m=ArchaeasHeatPlotBP.m[ArchaeasHeatPlotBP.m$variable %in% MetFam$Enzyme,]
ggplot(MetFam_BP.m, aes(x=MetFam_BP.m$variable, y=MetFam_BP.m$value, fill=Order))+ labs(x = "Metabolic E4P_AMINO_ACIDS Families", y = "Copies on Archaeabacteria Genomes",text = element_text(size=12)) + geom_boxplot() + facet_grid(. ~ Phylum)+theme_bw() +theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6), legend.position = "bottom")
#+ geom_jitter(aes(color=MetFam_BP.m$Phylum))


MetFam=subset(ArchaeasCentral,Pathway=="PYR_THR_AA")
MetFam_BP.m=ArchaeasHeatPlotBP.m[ArchaeasHeatPlotBP.m$variable %in% MetFam$Enzyme,]
ggplot(MetFam_BP.m, aes(x=MetFam_BP.m$variable, y=MetFam_BP.m$value, fill=Order))+ labs(x = "Metabolic Families on PYR_THR_AA pathway ", y = "Copies on Archaeabacteria Genomes",text = element_text(size=12)) + geom_boxplot() + facet_grid(. ~ Phylum)+theme_bw() +theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6), legend.position = "bottom")
#+ geom_jitter(aes(color=MetFam_BP.m$Phylum))

#ggsave("chapter3/expansion_plotArchaeas.pdf", plot = expansion_plotArchaea,height = 8, width = 7)

```

\clearpage 


<!----------------------Expansions Box plot END-------------------------------------------------------------------->
<!------------------------------------------------------------------------------------------------s--------------->




## Central pathway expansions  
Heat plot of central pathways expansions, Needs to be phylogenetically sorted.  
```{r ArchaeaHeatPlots, warnings = FALSE, message = FALSE, echo=FALSE,CACHE=TRUE}
### ME falta ordernarlos filogeneticamente
ArchaeasHeatPlot <- read.table("chapter3/ArchaeasHeatPlot", header=TRUE, sep="\t")
ArchaeasHeatPlot$order<-c(1:nrow(ArchaeasHeatPlot))
ArchaeasHeatPlot$RastId <- with(ArchaeasHeatPlot,reorder(ArchaeasHeatPlot$RastId, ArchaeasHeatPlot$order))
ArchaeasHeatPlot.m <- melt(ArchaeasHeatPlot)
ArchaeasHeatPlot.m<- ddply(ArchaeasHeatPlot.m, .(variable), transform,rescale=scale(value))  ## rescale escala toda la matriz, scale por columnas
HP_Archaeas_Taxa<-merge(ArchaeasHeatPlot,ArchaeasTaxa,by.x = "RastId",by.y = "RastId")
HP_Archaeas_Taxa2<-subset(HP_Archaeas_Taxa,Order='Chroococcales')

HP_Archaeas_Taxa<-HP_Archaeas_Taxa[order(HP_Archaeas_Taxa$Order,HP_Archaeas_Taxa$Family),]
HP_Archaeas_Taxa.m<- melt(HP_Archaeas_Taxa)
HP_Archaeas_Taxa.m<- ddply(HP_Archaeas_Taxa.m, .(variable), transform,rescale=scale(value))  ## rescale escala toda la matriz, scale por columnas

Archaeas_heatplot <- ggplot(HP_Archaeas_Taxa.m, aes(HP_Archaeas_Taxa.m$variable, HP_Archaeas_Taxa.m$RastId))+ geom_tile(aes(fill = HP_Archaeas_Taxa.m$rescale), colour = "white") + scale_fill_gradientn(colours = hm.palette(100),na.value = "gray")+ theme(text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )

ggplot(HP_Archaeas_Taxa.m, aes(HP_Archaeas_Taxa.m$variable, HP_Archaeas_Taxa.m$RastId))+ geom_tile(aes(fill = HP_Archaeas_Taxa.m$rescale), colour = "white") + scale_fill_gradientn(colours = hm.palette(100),na.value = "gray")+ theme(text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )

ggsave("chapter3/ArchaeasHeatPlot.pdf", plot = Archaeas_heatplot,height = 12, width = 6)

```

```{r ArchaeaHeatPlots_label, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter3/ArchaeasHeatPlot.pdf", caption = "Archaeas Heatplot",scale=.6,label = "ArchaeaPlot", type = "figure")
```
Here is a reference to the HeatPlot: `r ref("ArchaeaPlot")`. 
\clearpage 

```{r ArchaeaHeaplot_summary,echo=FALSE}
#summary(ArchaeasHeatPlot.m)
#table(ArchaeasHeatPlot.m$scale >=0,Deltas$DeltaPRO>=0)

```

<!---------------END EvoMining heatmap----------------------------------------------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->





## Genome Size correlations
### Correlation between genome size and AntiSMASH products  
```{r ANTISMASH_GenomeSize_Correlation_by_ORG_Archaeas,echo=FALSE}
ArchaeasEXP_SMASH<-merge(ArchaeasHeatPlot,ArchaeasSMASH,by.x="RastId",by.y="RastId")
ArchaeasEXP_SMASH2<-merge(ArchaeasTaxa, ArchaeasEXP_SMASH, by.x="RastId",by.y="RastId") 
Archaeassn1<-qplot(Size, TOTAL.y, data = ArchaeasEXP_SMASH2,colour=Order)+theme_bw()+ theme(legend.position	="bottom",plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
ggsave("chapter3/ArchaeasSMASHvsSizebyOrder.pdf", plot = Archaeassn1,height = 8, width = 6)

Archaeassn2<-qplot(Size, TOTAL.y, data = ArchaeasEXP_SMASH2,colour=Order)+theme_bw()+facet_grid(ArchaeasEXP_SMASH2$Order~.,scales="free")+ theme(legend.position	="bottom",plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
ggsave("chapter3/ArchaeasSMASHvsSizeGridOrder.pdf", plot = Archaeassn2,height = 8, width = 6)

```

Genome size vs Total antismash cluster coloured by order  
```{r ArchaeasSMASHvsSizebyOrder_label, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter3/ArchaeasSMASHvsSizebyOrder.pdf", caption = "Correlation between Archaeas genome size and antismash Natural products detection colored by Order",scale=.6,label = "ArchaeasSMASHvsSizebyOrder", type = "figure")
```
Here is a reference to Genome size vs Total antismash cluster: `r ref("ArchaeasSMASHvsSizebyOrder")`. 
\clearpage

Genome size vs Total antismash cluster detected splitted by order  
```{r ArchaeasSMASHvsSizeGridOrder_label, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter3/ArchaeasSMASHvsSizeGridOrder.pdf", caption = "Correlation between Archaeas genome size and antismash Natural products detection grided by Order",scale=.6,label = "ArchaeasSMASHvsSizeGridOrder", type = "figure")
```

Here is a reference to Correlation between genome size and antismash Natural products detection grided by Order plot: `r ref("ArchaeasSMASHvsSizeGridOrder")`. 
\clearpage 

<!--------------------------------END size vs AntiSMASH correlation--------------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->






<!-------------------------------- size vs Expansions correlation--------------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->
### Correlation between genome size and Central pathway expansions  
```{r ArchaeasANTISMASH_GenomeSize_Correlation_by_ORG,echo=FALSE}

ArchaeasEXP_TAXA<-merge(ArchaeasHeatPlot,ArchaeasTaxa,by.x="RastId",by.y="RastId")

Archaeassn4<-qplot(Size, TOTAL, data = ArchaeasEXP_TAXA,colour=Order)+theme_bw()+ theme(legend.position	="bottom",plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
ggsave("chapter3/ArchaeasSizevsExpansionsbyOrder.pdf", plot = Archaeassn4,height = 6, width = 6)

Archaeassn5<-qplot(Size, TOTAL, data = ArchaeasEXP_TAXA,colour=Order)+theme_bw()+facet_grid(ArchaeasEXP_TAXA$Order~.,scales="free")+ theme(legend.position	="bottom",plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
ggsave("chapter3/ArchaeasSizevsExpansionsGridbyOrder.pdf", plot = Archaeassn5,height = 6, width = 6)

## Looking for color modifications
#sn + scale_color_brewer(palette="Dark2(100)")
#n + scale_color_brewer(palette=hm.palette(10))
## sn+ scale_color_hue(l=40, c=35)
```
Genome size vs Total central pathway expansion coloured by order  
```{r ArchaeasSizevsExpansionsbyOrder_label2, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter3/ArchaeasSizevsExpansionsbyOrder.pdf", caption = "Correlation between Archaeas genome size and central pathway expansions ",label = "ArchaeasSizevsExpansionsbyOrder", type = "figure")
```
Here is a reference to the size vs Total central pathway expansion plot: `r ref("ArchaeasSizevsExpansionsbyOrder")`. 
\clearpage 


Genome size vs Total central pathway expansion grided by order  
```{r ArchaeasSizevsExpansionsGridbyOrder_label2, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter3/ArchaeasSizevsExpansionsGridbyOrder.pdf", caption = "Correlation between Archaeas genome size and central pathway expansions grided by order",label = "ArchaeasSizevsExpansionsGridbyOrder", type = "figure")
```
Here is a reference to the Genome size vs Total central pathway expansion grided by order plot: `r ref("ArchaeasSizevsExpansionsGridbyOrder")`. 
\clearpage 

<!--------------------------------------------------------------------------------------------------------------------->


<!---muy bien para el suplemenario-->
Correlation between genome size and each of the central pathway families.
Data are coloured by metabolic family instead of coloured by taxonomical order. This treatment allows to answer how differente metabolic families grows when genome size grow.  
Also I want to add form given by taxonomical order.

```{r ArchaeaANTISMASH_GenomeSize_Correlation_by_ORG,message=FALSE, echo=FALSE}
ArchaeasHeatPlot$RastId <- with(ArchaeasHeatPlot, reorder(ArchaeasHeatPlot$RastId, ArchaeasHeatPlot$TOTAL))
ArchaeasHeatPlot.m <- melt(ArchaeasHeatPlot)
ArchaeasEXP_TAXA<-merge(ArchaeasHeatPlot.m,ArchaeasTaxa,by.x="RastId",by.y="RastId")

Archaeasn6<-qplot(Size, value, data = ArchaeasEXP_TAXA,colour=variable,shape=Order)+theme_bw()+ theme(legend.position	="none",legend.text=element_text(size=6),plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
ggsave("chapter3/ArchaeasSizevsExpansionsbyMetabolicFamily.pdf", plot = Archaeasn6,height = 6, width = 8)

Archaeasn7<-qplot(Size, value, data = ArchaeasEXP_TAXA,colour=variable)+theme_bw()+facet_grid(ArchaeasEXP_TAXA$variable~.,scales="free")+ theme(legend.position	="none",plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
ggsave("chapter3/ArchaeasExpansionsbyMetabolicFamilyGrid.pdf", plot = Archaeasn7,height = 6, width = 12)

## Looking for color modifications
#sn + scale_color_brewer(palette="Dark2(100)")
#n + scale_color_brewer(palette=hm.palette(10))
## sn+ scale_color_hue(l=40, c=35)
```

Genome size vs Total central pathway expansion coloured by metabolic Family 
```{r ArchaeasSizevsExpansionsbyMetabolicFamily_label2, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter3/ArchaeasSizevsExpansionsbyMetabolicFamily.pdf", caption = "Correlation between Archaeas Genome size vs Total central pathway expansion coloured by metabolic Family",scale=.6,label = "ArchaeasSizevsExpansionsbyMetabolicFamily", type = "figure")
```
Here is a reference to the Genome size vs Total central pathway expansion coloured by metabolic Family plot: `r ref("ArchaeasSizevsExpansionsbyMetabolicFamily")`. 
\clearpage 

Future Work: Genome size vs Total central pathway expansion grided by metabolic Family 
For clarity I need to also grid and group by Metabolic Pathway
```{r ArchaeasExpansionsbyMetabolicFamilyGrid_label2, results = "asis",message=FALSE,echo=FALSE}
#label(path = "chapter3/ArchaeasExpansionsbyMetabolicFamilyGrid.pdf", caption = "Correlation between Archaeas Genome size vs Total central pathway expansion grided by metabolic Family ",label = "ArchaeasExpansionsbyMetabolicFamilyGrid", type = "figure")
```
Here is a reference to Genome size vs Total central pathway expansion grided by metabolic Family  plot: `r ref("ArchaeasExpansionsbyMetabolicFamilyGrid")`. 
\clearpage 


<!----------------------------END - size vs Expansions correlation--------------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->






<!----------------------------Natural Products vs Expansions correlation--------------------------------------------------->
<!--------------------------------------------------------------------------------------------------------------->

## Natural products  

### Natural products recruitments from EvoMining heatplot
We can see natural products recruitment after central pathways expansions colored by their kingdom.  
Natural products recruited by metabolic family, colored by phylogenetic origin.
```{r ArchaeasNP_recruitment_taxa,echo=FALSE,message=FALSE,warning=FALSE}
ArchaeasRecbyKing<-qplot(x=Central, data=ArchaeasNp, geom="bar", fill=kingdom, position="stack")+theme_bw()+ theme(legend.position	="bottom",text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )
ggsave("chapter3/ArchaeasRecruitmentsbyKingdom.pdf", plot = ArchaeasRecbyKing,height = 6, width = 12)

ArchaeasRecbytaxa<-qplot(x=Central, data=ArchaeasNp, geom="bar", fill=taxa, position="stack")+theme_bw()+ theme(legend.position	="bottom",text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )
ggsave("chapter3/ArchaeasRecruitmentsbyTaxa.pdf", plot = ArchaeasRecbytaxa,height = 6, width = 12)

```


Recruitments after central pathways expansions coloured by Kingdom
```{r ArchaeasRecruitmentsbyKingdom_label2, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter3/ArchaeasRecruitmentsbyKingdom.pdf", caption = "Archaeas Recruitmens on central families coloured by kingdom",scale=.6,label = "ArchaeasRecruitmentsbyKingdom", type = "figure")
```
Here is a reference to Recruitments after central pathways expansions colourd by Kingdom plot: `r ref("ArchaeasRecruitmentsbyKingdom")`. 

\clearpage 
Recruitments after central pathways expansions colourd by taxonomy
```{r ArchaeasRecruitmentsbyTaxa_label2, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter3/ArchaeasRecruitmentsbyTaxa.pdf", caption = "Archaeas Recruitmens on central families coloured by taxonomy",scale=.5,label = "ArchaeasRecruitmentsbyTaxa", type = "figure")
```
Here is a reference to Recruitments after central pathways expansions colourd by taxa plot: `r ref("ArchaeasRecruitmentsbyTaxa")`. 
\clearpage 




## Archaeas AntiSMASH 
```{r ArchaeasDiversity,echo=FALSE,message=FALSE,warning=FALSE}
## merge tables
Archaeasm1 <- merge(ArchaeasTaxa, ArchaeasSMASH, by.x = "RastId", by.y = "RastId")
Archaeasm1.m <- melt(Archaeasm1, id = c("RastId","Name","SuperPhylum","Phylum","Class","Order","Family","Contigs","Size","RastNo"))
Archaeasm2.m<-subset(Archaeasm1.m,value>0)

ArchaeasDiversity<-qplot(x=Order, data=Archaeasm1.m, geom="bar", fill=Family, position="stack")+theme_bw()+ theme(text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )
#table(m1.m$Order>0,m1.m$Family>0)
#qplot(x=clarity, data=diamonds, geom="bar", fill=cut, position="stack")
ggsave("chapter3/ArchaeasDiversity.pdf", plot =ArchaeasDiversity,height = 6, width = 14)

ArchaeasSmash<-qplot(x=Order, data=Archaeasm2.m, geom="bar", fill=variable, position="stack",colours = hm.palette(100000))+theme_bw()+ theme(legend.position="none",text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )
#table(m1.m$Order>0,m1.m$Family>0)
ggsave("chapter3/ArchaeasSmash.pdf", plot = ArchaeasSmash,height = 6, width = 14)
```
 
Taxonomical diversity on Archaeasbacteria Data
```{r ArchaeasDiversity_label2, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter3/ArchaeasDiversity.pdf", caption = "Archaeas Diversity",scale=.6,label = "ArchaeasDiversity", type = "figure")
```
Here is a reference to Recruitments after central pathways expansions colourd by taxa plot: `r ref("ArchaeasDiversity")`. 
\clearpage

Smash diversity
```{r ArchaeasSmash_label2, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter3/ArchaeasSmash.pdf", caption = "Archaeas Smash Taxonomical Diversity",scale=.5,label = "ArchaeasSmash", type = "figure")
```
Here is a reference to Recruitments after central pathways expansions colourd by taxa plot: `r ref("ArchaeasSmash")`. 
\clearpage



### AntisSMASH vs Central Expansions  
Is it a correlation between pangenome grow and central pathways expansions?  
```{r ArchaeasANTISMASH_Expansion_Correlation_by_ORG,message=FALSE, echo =FALSE}
ArchaeasEXP_SMASH<-merge(ArchaeasHeatPlot,ArchaeasSMASH,by.x="RastId",by.y="RastId")
ArchaeasEXP_SMASH2<-merge(ArchaeasTaxa, ArchaeasEXP_SMASH, by.x="RastId",by.y="RastId") 
#qplot(Total.x, Total.y, data = EXP_SMASH, colour=Order)+theme_bw()+
Archaeassn8<-qplot(TOTAL.x, TOTAL.y, data = ArchaeasEXP_SMASH2,colour=Order)+theme_bw()+ theme(legend.position	="bottom",legend.text=element_text(size=10),plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 10),axis.text.y=element_text(size = 7))
ggsave("chapter3/ArchaeasSMASHvsExpansionsbyOrder.pdf", plot = Archaeassn8,height = 12, width = 6)

Archaeassn9<-qplot(TOTAL.x, TOTAL.y, data = ArchaeasEXP_SMASH2,colour=Order)+theme_bw()+facet_grid(ArchaeasEXP_SMASH2$Order~.,scales="free")+ theme(legend.position	="bottom",legend.text=element_text(size=10),plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
ggsave("chapter3/ArchaeasSMASHvsExpansionsbyOrderGRID.pdf", plot = Archaeassn9,height = 10, width = 6)

## Looking for color modifications
#sn + scale_color_brewer(palette="Dark2(100)")
#n + scale_color_brewer(palette=hm.palette(10))
## sn+ scale_color_hue(l=40, c=35)
```
Total central pathway expansions by genome vs Total antismash cluster detected coloured by order  
```{r ArchaeasSMASHvsExpansionsbyOrder_label, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter3/ArchaeasSMASHvsExpansionsbyOrder.pdf", caption = "Correlation between Archaeas central pathway expansions and antismash Natural products detection",scale=.5,label = "ArchaeasSMASHvsExpansionsbyOrder", type = "figure")
```
Here is a reference to the expansions vs antismash NP's clusters plot: `r ref("ArchaeasSMASHvsExpansionsbyOrder")`. 
\clearpage 

Total central pathway expansions by genome vs Total antismash cluster detected splitted by order  
```{r ArchaeasSMASHvsExpansionsbyOrderGRID_label, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter3/ArchaeasSMASHvsExpansionsbyOrderGRID.pdf", caption = "Correlation between Archaeas central pathway expnasions and antismash Natural products detection",scale=.5,label = "ArchaeasSMASHvsExpansionsbyOrderGRID", type = "figure")
```
Here is a reference to the expansions vs antismash NP's clusters splitted by order plot `r ref("ArchaeasSMASHvsExpansionsbyOrderGRID")`. 
\clearpage 

<!------------------------------------------------------------------------------------------------------------->




AntisMAsh vs Expansions by taxonomic Family
```{r ArchaeasANTISMASH_Expansion_Correlation_by_Family,echo=FALSE,warning=FALSE}
ArchaeasEXP_SMASH<-merge(ArchaeasHeatPlot,ArchaeasSMASH,by.x="RastId",by.y="RastId")
ArchaeasEXP_SMASH2<-merge(ArchaeasTaxa, ArchaeasEXP_SMASH, by.x="RastId",by.y="RastId") 

Archaeasnpf<-qplot(TOTAL.x, TOTAL.y, data = ArchaeasEXP_SMASH2,colour=Family)+theme_bw()+facet_grid(ArchaeasEXP_SMASH2$Order~.,scales="free")+
        theme( plot.title = element_text(size = 14, face = "bold"),
              text = element_text(size = 12),
              axis.title = element_text(face="bold"),
              axis.text.x=element_text(size = 12),
              axis.text.y=element_text(size = 7))
ggsave("chapter3/Archaeasnpf.pdf", plot = Archaeasnpf,height = 12, width = 6)

#facet_grid(EXP_SMASH$Order~.,scales="free")+
```

Natural products colured by family
```{r Archaeasnpf_label, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter3/Archaeasnpf.pdf", caption = "Archaeas Natural products by family",scale=.6,label = "Archaeasnpf", type = "figure")
```
Here is a reference to the Natural products colured by family
 plot `r ref("Archaeasnpf")`. 
\clearpage 

<!------------------------------------------------------------------------------------------------------------->

##Selected trees from EvoMining
Phosphoribosyl_isomerase_3 family  
Figure from EvoMining

```{r ArchaeasPhosphoribosyl_isomerase_evo_tree, results = "asis",message=FALSE, echo=FALSE}
label(path = "chapter3/tree41.png", caption = "Phosphoribosyl isomerase A EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "Phosphoribosyl_isomerase_A_evo_tree", type = "figure")

label(path = "chapter3/tree42.png", caption = "Phosphoribosyl isomerase other EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "Phosphoribosyl_isomerase_other_evo_tree", type = "figure")

label(path = "chapter3/tree65.png", caption = "Phosphoribosyl anthranilate isomerase EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "Phosphoribosylanthranilate_isomerase_evo_tree", type = "figure")

```
\clearpage 

## 
Other possible databases Archaeal signatures *set of protein-encoding genes that function uniquely within the Archaea; most signature proteins
have  no  recognizable  bacterial  or  eukaryal  homologs*  [@graham_archaeal_2000]
## Footnotes and Endnotes

You might want to footnote something. ^[footnote text] The footnote will be in a smaller font and placed appropriately. Endnotes work in much the same way. More information can be found about both on the CUS site or feel free to reach out to <data@reed.edu>.

## Bibliographies

Of course you will need to cite things, and you will probably accumulate an armful of sources. There are a variety of tools available for creating a bibliography database (stored with the .bib extension).  In addition to BibTeX suggested below, you may want to consider using the free and easy-to-use tool called Zotero.  The Reed librarians have created Zotero documentation at <http://libguides.reed.edu/citation/zotero>.  In addition, a tutorial is available from Middlebury College at <http://sites.middlebury.edu/zoteromiddlebury/>.

_R Markdown_ uses _pandoc_ (<http://pandoc.org/>) to build its bibliographies.  One nice caveat of this is that you won't have to do a second compile to load in references as standard \LaTeX\ requires. To cite references in your thesis (after creating your bibliography database), place the reference name inside square brackets and precede it by the "at" symbol.  For example, here's a reference to a book about worrying:  [@Molina1994].  This `Molina1994` entry appears in a file called `thesis.bib` in the `bib` folder.  This bibliography database file was created by a program called BibTeX.  You can call this file something else if you like (look at the YAML header in the main .Rmd file) and, by default, is to placed in the `bib` folder.  

For more information about BibTeX and bibliographies, see our CUS site (<http://web.reed.edu/cis/help/latex/index.html>)^[@reedweb2007]. There are three pages on this topic:  _bibtex_ (which talks about using BibTeX, at <http://web.reed.edu/cis/help/latex/bibtex.html>), _bibtexstyles_ (about how to find and use the bibliography style that best suits your needs, at <http://web.reed.edu/cis/help/latex/bibtexstyles.html>) and _bibman_ (which covers how to make and maintain a bibliography by hand, without BibTeX, at <http://web.reed.edu/cis/help/latex/bibman.html>). The last page will not be useful unless you have only a few sources.

If you look at the YAML header at the top of the main .Rmd file you can see that we can specify the style of the bibliography by referencing the appropriate csl file.  You can download a variety of different style files at <https://www.zotero.org/styles>.  Make sure to download the file into the csl folder.

##### Tips for Bibliographies

- Like with thesis formatting, the sooner you start compiling your bibliography for something as large as thesis, the better. Typing in source after source is mind-numbing enough; do you really want to do it for hours on end in late April? Think of it as procrastination.
- The cite key (a citation's label) needs to be unique from the other entries.
- When you have more than one author or editor, you need to separate each author's name by the word "and" e.g. `Author = {Noble, Sam and Youngberg, Jessica},`.
- Bibliographies made using BibTeX (whether manually or using a manager) accept \LaTeX\ markup, so you can italicize and add symbols as necessary.
- To force capitalization in an article title or where all lowercase is generally used, bracket the capital letter in curly braces.
- You can add a Reed Thesis citation^[@noble2002] option. The best way to do this is to use the phdthesis type of citation, and use the optional "type" field to enter "Reed thesis" or "Undergraduate thesis." 

## Anything else?

If you'd like to see examples of other things in this template, please contact the Data @ Reed team (email <data@reed.edu>) with your suggestions. We love to see people using _R Markdown_ for their theses, and are happy to help.

