---
header-includes:
- \usepackage{graphicx,latexsym}
- \usepackage{amssymb,amsthm,amsmath}
- \usepackage{longtable,booktabs,setspace}
output: html_document
---

<!--- Cyanobacteria-->

```{r include_packages_5, include = FALSE,echo=FALSE}
# This chunk ensures that the reedtemplates package is
# installed and loaded. This reedtemplates package includes
# the template files for the thesis and also two functions
# used for labeling and referencing
if(!require(devtools))
  install.packages("devtools", repos = "http://cran.rstudio.com")
if(!require(dplyr))
    install.packages("dplyr", repos = "http://cran.rstudio.com")
if(!require(ggplot2))
    install.packages("ggplot2", repos = "http://cran.rstudio.com")
if(!require(reedtemplates)){
  library(devtools)
  devtools::install_github("ismayc/reedtemplates")
  }
library(reedtemplates)
flights <- read.csv("data/flights.csv")
```


# Cyanobacteria EvoMining Results {#ref_labels}
Cyanobacteria is phylum {Referencia de luis}  

## Tables

----------------------------------------------------------------------------------
  Factors                    Correlation between Parents & Child     
------------------------- ----------------------------------------- --------------
  GenomeDB                                  1245                         
  
  Families                                  65                          
------------------------- ----------------------------------------- --------------
Table: Families on Cyanobacteria \label{tab:inher}

\clearpage
<!-- clearpage ends the page, and also dumps out all floats.Floats are things like tables and figures. -->

```{r loading_tables_5,message=FALSE,echo=FALSE}
library(ggplot2)
library( reshape )
library(plyr )
library(RColorBrewer)
library(scales)
library(plyr)
library(RColorBrewer)
hm.palette <- colorRampPalette(rev(brewer.pal(11, 'Spectral')), space='Lab')  

#setwd("/home/nelly/GIT/phd/Thesis/tesis/chap5_files")
CyanosCentral <- read.table("chapter5/CyanosCentral", header=TRUE, sep="\t")
CyanosHeatPlot <- read.table("chapter5/CyanosHeatPlot", header=TRUE, sep="\t")
CyanosNp <- read.table("chapter5/CyanosNp", header=TRUE, sep="\t")
CyanosSMASH <- read.table("chapter5/CyanosSMASH", header=TRUE, sep="\t")
CyanosTaxa <- read.table("chapter5/CyanosTaxa", header=TRUE, sep="\t")

```

<!--
1. De Np's Quiero ver cantidad y diversidad de reclutamientos, a cada familia metabolica
   cuantos tiene y de donde provienen 
   
2. De antismash 
   Familia taxonomica vs clusters
   cluster interesante por familia taxonomica
   Voy a usar Cyanos Taxa tambien para ver esos reclutamientos de donde provienen

2. De HeatPlot quiero ordenarlo por filogenia 
Heat plot Familia taxonomica, 
boxplot reclutamientos para cada familia metabolica

fill <- "#4271AE"
lines <- "#1F3552"
ggplot(mydata110, aes(x=MName, y=delta,fill=TipeAc)) + geom_boxplot()+  scale_y_continuous(name = "Difference between old activity\nand new activity by step")+scale_x_discrete(name = "Mutation name")+theme_bw()  +
        theme(plot.title = element_text(size = 14, family = "Arial", face = "bold"),
              text = element_text(size = 12, family = "Arial"),
              axis.title = element_text(face="bold"),
              axis.text.x=element_text(size = 11),
              legend.position = "bottom") +facet_grid(mydata110$TipeAc~.,scales="free")
       

y hacer heatplots separados por ruta metabolica
4. Inlcuir algunos arboles

-->
## Natural products recruitments
We can see natural products origin by kingdom
Natural products recruited by metabolic family, colored by phylogenetic origin.
```{r NP_recruitment_taxa_cyano}
qplot(x=Central, data=CyanosNp, geom="bar", fill=kingdom, position="stack")+theme_bw()+ theme(text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )

```


## Cyanobacterias AntiSMASH 
Philogenetical diversity on Cyanobacterial dataset  
```{r CyanoDiversity}
#View(diamonds)
## merge tables
m1 <- merge(CyanosTaxa, CyanosSMASH, by.x = "RastId", by.y = "RastIds")
m1.m <- melt(m1, id = c("RastId", "Name","Phylum","Class","Order","Family","Contigs","Size","RastNo"))

qplot(x=Order, data=m1.m, geom="bar", fill=Family, position="stack")+theme_bw()+ theme(text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )
table(m1.m$Order>0,m1.m$Family>0)
#qplot(x=clarity, data=diamonds, geom="bar", fill=cut, position="stack")

```

## Figures
Central pathway expansions  

```{r HeatPlots, warnings = FALSE, message = FALSE, echo=FALSE}
CyanosHeatPlot$RastId <- with(CyanosHeatPlot, reorder(CyanosHeatPlot$RastId, CyanosHeatPlot$TOTAL))
CyanosHeatPlot.m <- melt(CyanosHeatPlot)
CyanosHeatPlot.m<- ddply(CyanosHeatPlot.m, .(variable), transform,rescale=scale(value))  ## rescale escala toda la matriz, scale por columnas

#heatplot <- ggplot(CyanosHeatPlot.m, aes(CyanosHeatPlot.m$variable, CyanosHeatPlot.m$RastId))+ geom_tile(aes(fill = CyanosHeatPlot.m$rescale), colour = "white") + scale_fill_gradient(low = "white", high = "steelblue")+ theme(text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )

heatplot <- ggplot(CyanosHeatPlot.m, aes(CyanosHeatPlot.m$variable, CyanosHeatPlot.m$RastId))+ geom_tile(aes(fill = CyanosHeatPlot.m$rescale), colour = "white") + scale_fill_gradientn(colours = hm.palette(100),na.value = "gray")+ theme(text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )


#+ scale_fill_gradientn(colours = hm.palette(100),na.value = "gray")
#+geom_text(aes(label=CyanosHeatPlot.m$rescale)) 
ggsave("chapter5/HeatPlot.pdf", plot = heatplot,height = 12, width = 6)

```

Ahora si la figura

```{r HeatPlots_label, results = "asis",message=FALSE}
label(path = "chapter5/HeatPlot.pdf", caption = "Cyanobacterial Heatplot",label = "CyanoPlot", type = "figure")
```
Here is a reference to the Reed logo: `r ref("CyanoPlot")`. 


\clearpage 

```{r Heaplot_summary}
#summary(CyanosHeatPlot.m)
#table(CyanosHeatPlot.m$scale >=0,Deltas$DeltaPRO>=0)

```
mm

<!-- starts a new page and stops trying to place floats such as tables and figures 
-->
```{r boxplot_CyanoHP,warnings = FALSE, message = FALSE}
expansion_plot<-ggplot(CyanosHeatPlot.m, aes(x=CyanosHeatPlot.m$variable, y=CyanosHeatPlot.m$rescale)) + geom_boxplot()+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6), legend.position = "bottom")

#ggplot(CyanosHeatPlot.m, aes(x=CyanosHeatPlot.m$variable, y=CyanosHeatPlot.m$rescale)) + geom_boxplot()+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6), legend.position = "bottom")
#+facet_grid(CyanosTaxa$class~.,scales="free")
#ggplot(CyanosHeatPlot.m, aes(x=CyanosHeatPlot.m$variable,y=CyanosHeatPlot.m$rescale,fill=CyanosHeatPlot.m$rescale)) 
# +facet_grid(CyanosHeatPlot.m$rescale~.,scales="free")

ggsave("chapter5/expansion_plot.pdf", plot = expansion_plot,height = 6, width = 7)
```
 why figure is not here?

\clearpage 
```{r boxplot_CyanoHP_label, results = "asis",message=FALSE}
label(path = "chapter5/expansion_plot.pdf", caption = "Expansions Boxplot",label = "Cyano_expansion_boxplot", type = "figure")
```
\clearpage 
Here is a reference to the expansion boxplot: `r ref("Cyano_expansion_boxplot")`.   


##Selected trees from EvoMining
##Lets analize Phosphoribosyl_isomerase_3 family  
Figure from EvoMining
\clearpage 

```{r Phosphoribosyl_isomerase_evo_tree, results = "asis",message=FALSE, echo=FALSE}
label(path = "chapter5/tree64.png", caption = "Phosphoribosyl isomerase EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "Phosphoribosyl_isomerase_evo_tree", type = "figure")
label(path = "chapter5/tree1.png", caption = "Phosphoribosyl isomerase EvoMiningtree", 
      scale = .25,
      angle = 180,
      label = "Phosphoribosyl_isomerase_evo_tree", type = "figure")

#label(path = "chapter5/tree64.png", caption = "Phosphoribosyl isomerase EvoMiningtree", 

```
\clearpage 

## Footnotes and Endnotes

You might want to footnote something. ^[footnote text] The footnote will be in a smaller font and placed appropriately. Endnotes work in much the same way. More information can be found about both on the CUS site or feel free to reach out to <data@reed.edu>.

## Bibliographies

Zotero.  The Reed librarians have created Zotero documentation at <http://libguides.reed.edu/citation/zotero>.  In addition, a tutorial is available from Middlebury College at <http://sites.middlebury.edu/zoteromiddlebury/>.

_R Markdown_ uses _pandoc_ (<http://pandoc.org/>) to build its bibliographies.  One nice caveat of this is that you won't have to do a second compile to load in references as standard \LaTeX\ requires. To cite references in your thesis (after creating your bibliography database), place the reference name inside square brackets and precede it by the "at" symbol.  For example, here's a reference to a book about worrying:  [@Molina1994].  This `Molina1994` entry appears in a file called `thesis.bib` in the `bib` folder.  This bibliography database file was created by a program called BibTeX.  You can call this file something else if you like (look at the YAML header in the main .Rmd file) and, by default, is to placed in the `bib` folder.  

For more information about BibTeX and bibliographies, see our CUS site (<http://web.reed.edu/cis/help/latex/index.html>)^[@reedweb2007]. There are three pages on this topic:  _bibtex_ (which talks about using BibTeX, at <http://web.reed.edu/cis/help/latex/bibtex.html>), _bibtexstyles_ (about how to find and use the bibliography style that best suits your needs, at <http://web.reed.edu/cis/help/latex/bibtexstyles.html>) and _bibman_ (which covers how to make and maintain a bibliography by hand, without BibTeX, at <http://web.reed.edu/cis/help/latex/bibman.html>). The last page will not be useful unless you have only a few sources.

If you look at the YAML header at the top of the main .Rmd file you can see that we can specify the style of the bibliography by referencing the appropriate csl file.  You can download a variety of different style files at <https://www.zotero.org/styles>.  Make sure to download the file into the csl folder.

##### Tips for Bibliographies

- You can add a Reed Thesis citation^[@noble2002] option. The best way to do this is to use the phdthesis type of citation, and use the optional "type" field to enter "Reed thesis" or "Undergraduate thesis." 

## Anything else?

If you'd like to see examples of other things in this template, please contact the Data @ Reed team (email <data@reed.edu>) with your suggestions. We love to see people using _R Markdown_ for their theses, and are happy to help.

