---
header-includes:
- \usepackage{amssymb,amsthm,amsmath}
- \usepackage{chemarr}
output: pdf_document
---

<!--
You can delete the header-includes (lines 3-5 above) if you like and also the chunk below since it is loaded in the skeleton.Rmd file.  They are included so that chap2.Rmd will compile by itself when you hit Knit PDF.
-->

```{r include_reedtemplates_2, include = FALSE}
# This chunk ensures that the reedtemplates package is installed and loaded
# This reedtemplates package includes the template files for the thesis and also
# two functions used for labeling and referencing
if(!require(devtools))
  install.packages("devtools", repos = "http://cran.rstudio.com")

if(!require(reedtemplates)){
  library(devtools)
  devtools::install_github("ismayc/reedtemplates")
  }
library(reedtemplates)
```


# PriA Family {#math-sci}

 * Julian simulation
 * Who has TrpF
 
```{r load_pkgs_chapter2, message = FALSE}
# List of packages required for this analysis
pkg <- c("dplyr", "ggplot2", "knitr", "devtools"," reshape","RColorBrewer")
# Check if packages are not installed and assign the
# names of the packages not installed to the variable new.pkg
new.pkg <- pkg[!(pkg %in% installed.packages())]
# If there are any packages in the list that aren't installed,
# install them
if (length(new.pkg))
  install.packages(new.pkg, repos = "http://cran.rstudio.com")
# Load packages
library(dplyr)
library(plyr)
library(reshape )
library(ggplot2)
library(knitr)
library(RColorBrewer)
hm.palette <- colorRampPalette(rev(brewer.pal(11, 'Spectral')), space='Lab')  
library(genstats) ## Next libraries are for coursera
library(devtools)
library(Biobase)
library(scales) # 
library(xlsx) # For save data on excel
```
 
 Strpetomyces genera has differences on PriA funcion. PriA 
## Math
Docking simulation were calculated for Streptomyces enzymes. TrpF enzymes from Strptomyces Mg1, Jonesia denitrificans, were added as controls


Procedures can be found at [Docking Protocols](https://github.com/tripplab/Docking/wiki)  

1.Phylogenetic Tree
39 Streptomyces sequences, as outgroup E coli, Arthrobacter Aurescens, Salmonella enterica and Acidimicrobium ferrooxidans PriA’s were included.


CORASON
PriA
All streptomyces have a partially conserved PriA cluster. CT34 has a secondary copy whose Best hit on NCBI is Lentzea’s PriA with 50% identity 98% coverage

TrpF1
TrpF1 queries gave hits with TrpC enzyme present on every Streptomyces, additionally  S rimosus, S coelicolor, S venezuelae and S. NRRL S-1813 had an extra copy. S rimosus TrpC vicinity has PKS and siderophore genes.

TrpF2
Conserved cluster with NRPS sequences flanking TrpF2

TrpF3
Non conserved cluster

TrpF4
purpeofuscus and S bikiniensis 
2. Heatmap
Additionally to the sequences selected by phylogeny, Jonesia denitrificans and Streptomyces sp Mg1 TrpF sequences were added as control .

```{r sessionInfo,echo=FALSE}

#vignette(package="genstats")
#vignette("01_06_three-tables")

# phenoData:
tmp <- read.csv("chapter2/ProteinData", row.names = 1,header=TRUE,sep="\t")
pdata <- AnnotatedDataFrame(tmp)

  # featureData:
tmp <- read.csv("chapter2/Substrate.data", row.names = 1,sep="\t")
fdata <- AnnotatedDataFrame(tmp)

# expression data:
tmp <- read.table("chapter2/EnsymeVsSubstrate.data",row.names = 1,header=TRUE,sep="\t")
m <- as.matrix(tmp)
#dim(m)
#class(m)
#colnames(m)
#rownames(m)
## Names should not start on numbers never
## create ExpressionSet object:
eset <- new("ExpressionSet", exprs = m, phenoData = pdata, featureData = fdata)

#pData(eset)
#fData(eset)
#pData(eset)
#fData(eset)
```
Substrate table  
```{r Substrates_table_out, results = "asis"}
table <- read.csv("chapter2/Substrate.data", row.names = 1,sep="\t")
kable(table,  caption = "Substrates \\label{tab:substrates}",caption.short = "Substrates ")
```
Subi a mazroca para mrBayes PriAjulian.nxs  
Baje el arbol PriAjulian.nxs  
Obtuve las coordenadas con el script
Hice el heatplot en R para obtener los colores 
Hice el heatplot en SVG con el script   

Observaciones Mauricio  
- falta el pie de figura
Substrate docking over enzymes from extended PriA families. Substrates 1-20  
  
- falta el dendograma arriba de los sustratos (indicar cuál es pra y cuál profar)   

- falta la escala de colores (con valores y unidades). Cuando agregues el dendograma de los sutratos va a quedar un cuadrante vacío en la parte superior izquierda, en ese espacio pon la escala de forma horizontal  


- intenta aumentar el contraste de color entre el valor mínimo y máximo de la escala  
Intente varias Figuras  

- repite las etiquetas de los sutratos abajo del heatmap  
Pendiende  

- hay inconsistencias en los nombres de la enzimas (eg, Ssvi y Svi, SMgl y Smgl, etc)  
Árbol nuevo sin datos faltantes  

- para el caso de las enzimas sin datos, usa negro en lugar de amarillo  
LISTO  

- "Controles TrpF/subTrpF" (debe estar en inglés), puede ser confuso poner dos etiquetas para la misma función  

```{r load_data_docking}
docking <- read.csv("chapter2/SmallHeat.data", header=TRUE, sep="\t")
EnzymeOrder=factor(docking$Enzima,levels=unique(docking$Enzima))
docking.m <- melt(docking,id = "Enzima")
docking.m <- ddply(docking.m, .(variable), transform,rescale = rescale(value))
docking.m$Enzima <- factor(docking.m$Enzima, levels = rev(docking.m$Enzima[order(EnzymeOrder)]))

ggplot(docking.m, aes(x=variable, y=Enzima)) + labs(x = "Substrates", y = "Enzymes",text = element_text(size=12))+ geom_raster(aes(fill=value))+scale_fill_gradient(low = "white", high = "steelblue",na.value = "black")+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6))

ggplot(docking.m, aes(x=variable, y=Enzima)) + labs(x = "Substrates", y = "Enzymes",text = element_text(size=12))+ geom_raster(aes(fill=value))+scale_fill_gradient(low = "white", high = "black",na.value = "black")+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6))

ggplot(docking.m, aes(x=variable, y=Enzima)) + labs(x = "Substrates", y = "Enzymes",text = element_text(size=12))+ geom_raster(aes(fill=value))+scale_fill_gradient(low = "white", high = "steelblue",na.value = "black")+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6))+ scale_fill_gradientn(colours = hm.palette(100),na.value = "gray")

## boxplot de los sustratos
ggplot(docking.m, aes(x=variable, y=Enzima)) + labs(x = "Substrates", y = "Enzymes",text = element_text(size=12))+ geom_raster(aes(fill=value))+scale_fill_gradient(low = "white", high = "steelblue",na.value = "black")+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6))

p<-ggplot(docking.m, aes(x=variable, y=Enzima)) + labs(x = "Substrates", y = "Enzymes",text = element_text(size=12))+ geom_raster(aes(fill=value))+scale_fill_gradient(low = "white", high = "steelblue",na.value = "black")+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6))
g <- ggplot_build(p)
g$data[[1]]["fill"]
g$data[[1]]
write.xlsx(g$data[[1]], "chapter2/mydata.xlsx")

ggplot(docking.m, aes(x=variable, y=value)) + labs(x = "Metabolic Families", y = "Copies on Actinobacteria Genomes",text = element_text(size=12)) + geom_boxplot()+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 6), legend.position = "bottom")

```

We next introduce a useful function in the `knitr` package for making nice tables in _R Markdown_ called `kable`.  It produces the \LaTeX\ code required to make the table and is much easier to use than manually entering values into a table by copying and pasting values into Excel or \LaTeX.  This again goes to show how nice reproducible documents can be!  There is no need to copy-and-paste values to create a table.  (Note the use of `results = "asis"` here which will produce the table instead of the code to create the table.  You'll learn more about the `\\label` later.)  The `caption.short` argument is used to include a shorter version of the title to appear in the List of Tables at the beginning of the document.

```{r docking_table_out, results = "asis"}
kable(docking,  caption = "Enzymes docking \\label{tab:docking}",caption.short = "Enzymes docking ")


```

We can further look into the properties of the largest value here for American Airlines Inc.  To do so, we can isolate the row corresponding to the arrival delay of 1539 minutes for American in our original `flights` dataset.

```{r max_props2}
#flights %>% dplyr::filter(arr_delay == 1539, carrier_name == "American Airlines Inc.") %>%
  #dplyr::select(-c(month, day, carrier, dest_name, hour, minute, carrier_name, arr_delay))
```

We see that the flight occurred on March 3rd and departed a little after 2 PM on its way to Dallas/Fort Worth.  Lastly, we show how we can visualize the arrival delay of all departing flights from Portland on March 3rd against time of departure.


```{r march3plot_2, fig.height = 3, fig.width = 6}
#flights %>% dplyr::filter(month == 3, day == 3) %>%
#  ggplot(aes(x = dep_time, y = arr_delay)) +geom_point()
```

<!--
--------------------------------------------------------------------------->

Genome size vs Total antismash cluster coloured by order  
```{r PriADocking, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter2/PriAHeatPot.png", caption = "Heat Plot PriA Streptomyces vs other subtrates",scale=.6,label = "PriADocking", type = "figure")
```

Docker simulation were calculated for Streptomyces enzymes  
Genome size vs Total antismash cluster coloured by order  
```{r TrpFDocking, results = "asis",message=FALSE,echo=FALSE}
label(path = "chapter2/TrpFHeatPlot.png", caption = "Heat Plot TrpF Streptomyces vs other subtrates",scale=.6,label = "TrpFDocking", type = "figure")
```

\TeX\ is the best way to typeset mathematics. Donald Knuth designed \TeX\ when he got frustrated at how long it was taking the typesetters to finish his book, which contained a lot of mathematics.  One nice feature of _R Markdown_ is its ability to read \LaTeX\ code directly.

If you are doing a thesis that will involve lots of math, you will want to read the following section which has been commented out. If you're not going to use math, skip over or delete this next commented section.


<!-- MATH and PHYSICS majors: Uncomment the following section -->
<!--
$$\sum_{j=1}^n (\delta\theta_j)^2 \leq {{\beta_i^2}\over{\delta_i^2 + \rho_i^2}}
\left[ 2\rho_i^2 + {\delta_i^2\beta_i^2\over{\delta_i^2 + \rho_i^2}} \right] \equiv \omega_i^2
$$

From Informational Dynamics, we have the following (Dave Braden):

After _n_ such encounters the posterior density for $\theta$ is

$$
\pi(\theta|X_1< y_1,\dots,X_n<y_n) \varpropto \pi(\theta) \prod_{i=1}^n\int_{-\infty}^{y_i}
   \exp\left(-{(x-\theta)^2\over{2\sigma^2}}\right)\ dx
$$

Another equation:

$$\det\left|\,\begin{matrix}%
c_0&c_1\hfill&c_2\hfill&\ldots&c_n\hfill\cr
c_1&c_2\hfill&c_3\hfill&\ldots&c_{n+1}\hfill\cr
c_2&c_3\hfill&c_4\hfill&\ldots&c_{n+2}\hfill\cr
\,\vdots\hfill&\,\vdots\hfill&
  \,\vdots\hfill&&\,\vdots\hfill\cr
c_n&c_{n+1}\hfill&c_{n+2}\hfill&\ldots&c_{2n}\hfill\cr
\end{matrix}\right|>0$$


Lapidus and Pindar, Numerical Solution of Partial Differential Equations in Science and
Engineering.  Page 54

$$
\int_t\left\{\sum_{j=1}^3 T_j \left({d\phi_j\over dt}+k\phi_j\right)-kT_e\right\}w_i(t)\ dt=0,
   \qquad\quad i=1,2,3.
$$

L\&P  Galerkin method weighting functions.  Page 55

$$
\sum_{j=1}^3 T_j\int_0^1\left\{{d\phi_j\over dt} + k\phi_j\right\} \phi_i\ dt
   = \int_{0}^1k\,T_e\phi_idt, \qquad i=1,2,3 $$

Another L\&P (p145)

$$
\int_{-1}^1\!\int_{-1}^1\!\int_{-1}^1 f\big(\xi,\eta,\zeta\big)
   = \sum_{k=1}^n\sum_{j=1}^n\sum_{i=1}^n w_i w_j w_k f\big( \xi,\eta,\zeta\big).
$$

Another L\&P (p126)

$$
\int_{A_e} (\,\cdot\,) dx dy = \int_{-1}^1\!\int_{-1}^1 (\,\cdot\,) \det[J] d\xi d\eta.
$$
-->

## Chemistry 101: Symbols

Chemical formulas will look best if they are not italicized. Get around math mode's automatic italicizing in \LaTeX\ by using the argument `$\mathrm{formula here}$`, with your formula inside the curly brackets.  (Notice the use of the backticks here which enclose text that acts as code.)

So, $\mathrm{Fe_2^{2+}Cr_2O_4}$ is written `$\mathrm{Fe_2^{2+}Cr_2O_4}$`.

<!--
The \noindent command below does what you'd expect:  it forces the current line/paragraph to not indent. This was done here to match the format of the LaTeX thesis PDF.
-->

\noindent Exponent or Superscript: $\mathrm{O^-}$

\noindent Subscript: $\mathrm{CH_4}$

To stack numbers or letters as in $\mathrm{Fe_2^{2+}}$, the subscript is defined first, and then the superscript is defined.

\noindent Angstrom: ${\AA}$

\noindent Bullet: CuCl $\bullet$ $\mathrm{7H_{2}O}$

\noindent Double Dagger: $\ddag$

\noindent Delta: $\Delta$

\noindent Reaction Arrows: $\longrightarrow$ or  $\xrightarrow{solution}$

\noindent Resonance Arrows: $\leftrightarrow$

\noindent Reversible Reaction Arrows: $\rightleftharpoons$ or $\xrightleftharpoons[ ]{solution}$ (the latter requires the `chemarr` \LaTeX\ package which is automatically loaded in this template)

### Typesetting reactions

You may wish to put your reaction in a figure environment, which means that \LaTeX\ will place the reaction where it fits and you can have a figure caption.  You'll see further description of this **R** `label` function in [](#ref_labels). (Note the use of the double backslash here as well as the `echo = FALSE` which hides the code from the output.)

```{r chem_reac_figure, results = "asis", echo = FALSE}
label(path = "$\\mathrm{C_6H_{12}O_6  + 6O_2} \\longrightarrow \\mathrm{6CO_2 + 6H_2O}$", 
      caption = "Combustion of glucose", 
      label = "comb-gluc", 
      type = "chem.reac")
```

### Other examples of reactions

$\mathrm{NH_4Cl_{(s)}}$ $\rightleftharpoons$ $\mathrm{NH_{3(g)}+HCl_{(g)}}$

\noindent $\mathrm{MeCH_2Br + Mg}$ $\xrightarrow[below]{above}$ $\mathrm{MeCH_2\bullet Mg \bullet Br}$

## Physics

Many of the symbols you will need can be found on the math page <http://web.reed.edu/cis/help/latex/math.html> and the Comprehensive \LaTeX\ Symbol Guide (<http://mirror.utexas.edu/ctan/info/symbols/comprehensive/symbols-letter.pdf>).

## Biology

You will probably find the resources at <http://www.lecb.ncifcrf.gov/~toms/latex.html> helpful, particularly the links to bsts for various journals. You may also be interested in TeXShade for nucleotide typesetting (<http://homepages.uni-tuebingen.de/beitz/txe.html>).  Be sure to read the proceeding chapter on graphics and tables.


```{r endsession}
sessionInfo()

```
