---
title: "EvoMining"
author: "Nelly Selem"
date: "December31, 2017"
output:
  word_document: default
  pdf_document: default
  html_document: default
nocite: |
  @cibrian-jaramillo_increasing_2016,@barona-gomez_what_2012,@cruz-morales_phylogenomic_2016,@medema_minimum_2015,@ziemert_evolution_2016,@medema_computational_2015
  @dufresne_algorithmique_2016@,@blin_recent_nodate,@kurtboke_revisiting_2017,@miller_interpreting_2017,@schniete_expanding_2017,@kim_recent_2017,@robertsen_toward_2017,@juarez-vazquez_evolution_nodate,@chavali_bioinformatics_nodate,@tracanna_mining_2017,@ren_breaking_2017,@choudhary_current_2017,@alanjary_antibiotic_2017,@chevrette_sandpuma:_2017,@wohlleben_antibiotic_2016,@weber_secondary_2016
bibliography: bib/thesis.bib
---


```{r setup, include=FALSE}
library(knitr)
knitr::opts_chunk$set(echo = TRUE)
```

```{r loading_tables_Actino,message=FALSE,echo=FALSE}
library(ggplot2)
library( reshape )
library(plyr ) 
library(RColorBrewer)
library(scales)
library(dplyr)
if(!require(devtools))
  install.packages("devtools", repos = "http://cran.rstudio.com")
if(!require(easyGgplot2))
  install_github("kassambara/easyGgplot2")
library(easyGgplot2)

hm.palette <- colorRampPalette(rev(brewer.pal(11, 'Spectral')), space='Lab')  
tropical=c('darkorange','dodgerblue','hotpink')
palette(tropical)


```

<!--codigo de evoMining display column even if only ceros, se despliega si le bajas el score a cero, pero quiero que se despliegue siempre-->
<!-- Plan
Figura 1 Figuras/EvoMiningPipeline.svg Metabolismo expansion
Figura 2 Figuras/GenomicDBS.svg Semana 2 Base genomica ->PriA
Figura 3 Figuras/TauDpseudomonas2.svg Base central -> TauD
Figura 4 Figuras/Backward.svg   Semana 1 AntiSMASH  
Figura 5 Figuras/PanCluster.svg  Semana 3  Poner los clusters de la nueva Familia  
-->

## EvoMining 2.0: A customizable computational pipeline for evolutionary reconstructions during genome mining  
Selem-Mojica Nelly, Cruz-Morales Pablo, Martínez-Guerrero Christian , ...,  and Barona-Gómez Francisco   							  
  
## Abstract  
<!-- Por qué son importantes los productos naturales  
Por qué es importante el genome mining en esta investigación  
Cómo se ha resuelto el problema hasta ahora  
Que aporta EvoMining  
Qué se hace en este estudio  
Cuáles son las perspectivas  -->
Microbial natural products has importance in human health and life. Due to the abundance of genomic and metagenomic data, new natural products research by genome mining is a growing field. Traditional genome mining  approaches explored bacterial genomes localizing marks of previously knwon secondary metabolism enzymes organized on biosynthetic gene clusters (BGCs). Here we present EvoMining a downloadable visual genome mining tool that incorporates evolution theory into genome mining. On EvoMining databases are customizable, its based on enzyme expansions not on BGCs. The advantage of this method is that every expanded enzyme family is a candidate to explore recruitments, and all prokatyiotic genome, even the unexplored Archaea kingdom. On this study EvoMining was applied to several database such as Cyanobacteria, Actinobacteria, Pseudomonas and Archaea studying expansions for enzyme families such as PriA,TauD and other enzymes recently recruited onto secondary metabolism. Finally the genomic plasticity of Streptomyces coelicolor known BGCs is explored generalizing/applying the open/Close pangenome approach to a BGCs. This Evolutionary methods open the door to discover not previously knwon chemical compounds at private genome collections and prioritize them according to their genomic plasticity.    
  
## Introduction  
<!--
- Productos Naturales  
- BGCs enzymas en los BGCs  
- software de priorizacion.  
- Bases de datos centrales en los que se ha explorado  Backward TauD,
- Bases de datos genómicas  TauD en Pseudomonas, Genomica de Archaea Cyanobacteria  
- Apertura y plasticidad de un genoma  
- Repetibilidad  y personalización permiten exploración.  
-->

Natural products are synthesized by biosynthetical gene clusters (BGCs) codified on the genome of a wide range of microorganisms. Enzymes that belong to a BGC can either be mainly restricted to secondary metabolism, or be a recent recruitment acting as accesory enzymes.  
With the genomic era and 500,000 prokaryotic genomes available at NCBI, there has been a oom of development of specilized genome mining software. Traditional approaches are based on recognize marks of enzymes devoted to secondary metabolism [@antismash], or domains [@clusterfinder] lattely Evolution [@arts nadine].   
On prokaryotic genomes enzyme families are expanded frequently either by duplication or by horizontal gene transfer and that this expansions are acting as evolutionary raw material being recruited into secondary metabolism to perform nobel chemical functionalities. A proof of concept of EvoMining idea was provided by the discovery of an arseno compound on Streptomyces coelicolor [@cruz-morales_phylogenomic_2016], nevertheless. 

Despite EvoMining analysis has recently being present on the natural products field [@blin_recent_nodate,@alanjary_antibiotic_2017,@ziemert_evolution_2016,@miller_interpreting_2017] EvoMining software has not been released, on this work we free EvoMining as a downloadable stand alone tool implemented on a docker container. EvoMining is free and open to all users and there is no login requirement. 
Despite Actinobacteria are great natural product producers [@omura] other microrganisms can be explored.  

Here we present the EvoMining expansions analysis using different genome-DB such as Actinobacteria, Cyanobacteria, Pseudomonas and Archaea. To enrich possibilities of central DB an example of what we called backward EvoMining was incorporated: BGCs from S coelicolor available at Mi-BIG were analyzed EvoMining backwards and all enzyme families expanded but not over represented were followed. 

Finally to prioritize which clusters possess more metabolite variations, assuming a link between genomic and metabolite plasticity we introduce the idea of classifying the saturation of a pangenome as open/closed pangenome measuring BGCs as open / closed BGC.    

##  Results and Discussion   

### Figure 1  EvoMining pipe-line   
 EvoMining is a visual genome mining tool that following evolutionary principles has the milestone of prioritize non standard secondary metabolite pathways. The algorithm follows enzyme families from central pathways on their recruitment as components of natural products biosynthetic gene clusters (BGCs) within a genomic database. 
  
Figure1  
![Pipeline](Figuras/EvoMiningPipeline.svg)  
  
EvoMining inputs are a (1) a custom genomic database (genomic-DB), (2) a central pathways database (central-DB) and (3) a natural product database (natural-DB) composed of genes that belongs to experimentally tested BGCs. 
 These three databases are provided and can be modified, replaced and expanded by the user. In this work genomic-DB are  collection of up to date genomes in RAST format from taxonomically related organisms such as Actinobacteria, Cyanobacteria, Pseudomonas and Archaea. Selection of this taxa obeys to the possibility of comparing well known NPs producing organisms such as Actinobacteria and Cyanobacteria in contrast with Archaea that has been poorly investigated. The central-DB contains nine central pathways from Actinobacteria previously curated [@barona-gomez_what_2012], plus an update of seed metabolic enzymes identified after manual curation congruent with the central EvoMining paradigm.  The natural-DB currently comprises all sequences that belongs to some BGCs from The Minimum Information about a Biosynthetic Gene cluster (MIBiG) [@medema_minimum_2015].   

As output EvoMining identifies on the genomic-DB those expanded families from the central-DB that has at least a recruited member onto the natural-DB, proceeding then to the reconstruction of the evolutionary history of the enzyme family. Given an enzyme from the central-DB, the product of EvoMining analysis is a color coded tree of the expanded enzyme family that provides information about the metabolic fate. Specifically, enzymes from central metabolism are differentiated from known Natural Products enzymes and those expansions with potential activity into secondary metabolism are emphasised as putative novel recruitments. Further analysis of these hits allows visualization of the genomic vicinity guiding to the discovery of novel BGCs. In addition to the updates associated to the workflow of EvoMining, the version to be released will include the possibility of defining the dynamics of the gene content of any given BGC to explore the chemical plasticity related to EvoMining hits. This allows to prioritize which clusters possess more metabolite variations, therefore unmasking biosynthetic darkmatter (Medema and Fischbach 2015, Blin et al. 2017). 

EvoMining code and components (blast, muscle, FastTree, newick utilities, Gblocks,apache and SVG perl module) are wrapped on the docker container  nselem/newevomining downloadable at the Docker hub. Code is available at at github: nselem/EvoMining and manual at https://github.com/nselem/EvoMining/wiki. EvoMining tool will allow researchers to examine their own genomes and their own enzyme families in the search of expansions involved on nobel secondary metabolism.  

EvoMining will identify those expanded families of the central-DB within the genomic-DB that has at least a recruited member onto the natural-DB, proceeding then to the reconstruction of the evolutionary history of the enzyme family. Given an enzyme from the central-DB, the product of EvoMining analysis is an interactive color coded tree of the enzyme expanded family where best bidirectional hits (BBH) of central-DB are differentiated from Natural Products members and those expansions close to a Natural Product sequence that are not BBH with central-DB enzymes are emphasised as putative nobel recruitments into secondary metabolism.   
EvoMining algorithm was used before with XXX bacteria XX central, nvertheless  
Genomic DB 
### Figure 2  Expansions on some databases   
El reclutamiento de una enzima en una ruta de metabolismo secundario 
The expansioness of a family depends on the clade.  
Expansions are correlated with genome side, at different velocities, not linearly, Archaea Bacteria are smaller in size until now, Cyanobacteria correlation is weaker. 

Archaea has central expansions and has BGCs  
  
### Archaea fits as EvoMining central DB  
During the decade between 1970 and 1980, Archaea was recognized as new life domain, a kingdom different from Bacteria and Eucarya in an exciting first great application of 16S phylogeny[@woese_phylogenetic_1977,@woese_are_1981] . Main differences between this kingdoms are that Archaeal DNA is not arranged in a nucleus as in Eucarya and Archaeal celular walls are not composed from peptidoglycans as in Bacteria. Archaeal proteins may be higlhy valuable to biotechnology industry for their great stability due to extreme temperature, PH and salt content conditions on Archeal habitats. Despite no Archaeal Natural products biosynthetic gene clusters (BGC's) has been reported on MiBIG, Archaea do have BGC's, some of them seems to be acquired by horizontal gene transfer (HGT) like methano nrps {search reference}. Other Archeal natural products known are archaeosins, Diketopiperazines, Acyl Homoserine Lactones, Exopolysaccharides, Carotenoids, Biosurfactants, Phenazines and Organic Solutes but this knowledge is not comparable to Bacterial BGC's knowledge[@charlesworth_untapped_2015].  
  
Natural products biosynthetic gene clusters search is actually performed using either *high-confidence/low-novelty or low-confidence/high-novelty* bioinformatic approaches [@medema_computational_2015]. High confidence methods compares query sequences with previously known BGC's such as nrps or PKS, examples of this algorithms are antiSMASH and clusterfinder [@_antismash_????]. EvoMining searches on expansions from central metabolic pathways enzyme families, it has been classified as low confidence/high novelty method. EvoMining has proved useful on Actinobacteria phylum where its use lead to Arseno-compounds discovery [@cruz-morales_phylogenomic_2016]. Also on Actinobacteria antiSMASH analysis on 1245 genomes found 774 different classes of natural products, the same analysis on 876 Archaeal genomes, a full kingdom, identifies only 35 BGC's classes. So either Archaea does not have natural products BGC's or this are not yet known. Next paragraph deals with a possible approach about how natural products BGC's can be find.  
  
Archaea resembled Bacteria in that Archaea uses horizontal gene transfer as a genic interchange mecanism, Archaeal genomes contains operons [@howland_surprising_2000] and in general there is introns absence{Reference to Computational Methods for Understanding Bacterial and Archaeal Genomes}. Archaeas do have introns, but they are mainly located on genes that encodes ribosomal and transfer RNA [@howland_surprising_2000]. General lack of introns allows automatic genome annotation, operons gene organization permits functional inference to a certain degree and HGT contribute to expansions on Archaeal genomes. Some phylum on Archaea has an open pangenome, and as we will show on this chapter some Archaea has central pathway expansions. 
Enzyme families from central pathways expansions, open pangenome and operon organization made EvoMining succesful on Actinobacteria, this lead us to think that evoMining is suitable to analize Archaeal genomes, even more since EvoMining is a method oriented to use evolution and its not entirelyy based on previous knowledge of BGC's sequences if evolutionary logic behave on Archaea as on bacteria, new BGC's classes may be found on Archaea.
  
EvoMining is a trade off between conserved known central metabolic function and enough expansions divergence on sequence and on clusters to divergence 

Beside the already tested 240 Actinobacteria database, other genomic database can be explored. On this case we tested exapnsions on Archaea Cyanobacteria, and Actinobacteria based on central metabolism from actinobacteria  

On cyanobacteria PriA an enzyme with no previously knwon secondary metabolism on Actinobacteria has been recruited into saxitoxin cluster.   

![GenomicDatabases](Figuras/GenomicDBS.svg)
2.1 Expansions same central  
```{r GenomicDatabase,warnings = FALSE, message = FALSE,echo=FALSE}
######################## Loading data  
ArchaeasCentral <- read.table("../chapter3/ArchaeasCentral", header=TRUE, sep="\t")
ArchaeasHeatPlot <- read.table("../chapter3/ArchaeasHeatPlot", header=TRUE, sep="\t")
ArchaeasNp <- read.table("../chapter3/ArchaeasNp", header=TRUE, sep="\t")
ArchaeasSMASH <- read.table("../chapter3/ArchaeasSMASH", header=TRUE, sep="\t")
ArchaeasTaxa <- read.table("../chapter3/ArchaeasTaxa", header=TRUE, sep="\t")
ArchaeasTaxa$Kingdom<-"Archea"
ArchaeasSMASH$Kingdom<-"Archea"

ActinosCentral <- read.table("../chapter4/ActinoCentral", header=TRUE, sep="\t")
ActinosHeatPlot <- read.table("../chapter4/ActinoHeatPlot", header=TRUE, sep="\t")
ActinosNp <- read.table("../chapter4/ActinoNp", header=TRUE, sep="\t")
ActinosSMASH <- read.table("../chapter4/ActinosSMASH", header=TRUE, sep="\t")
ActinosTaxa <- read.table("../chapter4/ActinoTaxa", header=TRUE, sep="\t")
ActinosTaxa$Kingdom<-"Actinos"
ActinosSMASH$Kingdom<-"Actinos"
ActinosTaxa$SuperPhylum<-"Actinos"

CyanosCentral <- read.table("../chapter5/CyanosCentral", header=TRUE, sep="\t")
CyanosHeatPlot <- read.table("../chapter5/CyanosHeatPlot", header=TRUE, sep="\t")
CyanosNp <- read.table("../chapter5/CyanosNp", header=TRUE, sep="\t")
CyanosSMASH <- read.table("../chapter5/CyanosSMASH", header=TRUE, sep="\t")
CyanosTaxa <- read.table("../chapter5/CyanosTaxa", header=TRUE, sep="\t")
CyanosTaxa$Kingdom<-"Cyanos"
CyanosSMASH$Kingdom<-"Cyanos"
CyanosTaxa$SuperPhylum<-"Cyanobacteria"
########################### End of loading data

#################################################################
## Joining data frames heat plot and taxa
TodosTaxa<-rbind(ArchaeasTaxa,CyanosTaxa,ActinosTaxa)

#names(ArchaeasHeatPlot)
intersect(names(CyanosHeatPlot),names(ActinosHeatPlot))

intersect(intersect(names(CyanosHeatPlot),names(ActinosHeatPlot)),names(ArchaeasHeatPlot))
ArHP<-ArchaeasHeatPlot[,intersect(intersect(names(CyanosHeatPlot),names(ActinosHeatPlot)),names(ArchaeasHeatPlot))]
CyHP<-CyanosHeatPlot[,intersect(intersect(names(CyanosHeatPlot),names(ActinosHeatPlot)),names(ArchaeasHeatPlot))]
AcHP<-ActinosHeatPlot[,intersect(intersect(names(CyanosHeatPlot),names(ActinosHeatPlot)),names(ArchaeasHeatPlot))]
TodosHP<-rbind(ArHP,CyHP,AcHP)
TodosEXP_TAXA<-merge(TodosHP,TodosTaxa,by.x="RastId",by.y="RastId")

#str(TodosHP)
#intersect(names(CyanosHeatPlot),names(ArchaeasHeatPlot))
#intersect(intersect(names(CyanosHeatPlot),names(ActinosHeatPlot)),names(ArchaeasHeatPlot))
####  End of joining dataframes
###########################################################################################

################# Graphs 


## 1) Genome Size vs central expansion by facets 
GenomeSizeVsCentralExpansion<-qplot(TodosEXP_TAXA$Size, TodosEXP_TAXA$TOTAL, data = TodosEXP_TAXA,colour=TodosEXP_TAXA$Order)+facet_grid(TodosEXP_TAXA$Kingdom~.,scales="free")+theme_bw()+ theme(legend.position	="none",plot.title = element_text(size = 14, face = "bold"),text = element_text(size = 12),axis.title = element_text(face="bold"),axis.text.x=element_text(size = 12),axis.text.y=element_text(size = 7))
####ggsave(file="GenomeSizeVsCentralExpansion.svg", plot=GenomeSizeVsCentralExpansion, width=30, height=8)
## End genome size vs central expansion  

## 2) Box plot Expansiones familia y point Genome 
TodosHP$RastId <- with(TodosHP, reorder(TodosHP$RastId, TodosHP$TOTAL))
TodosHP.m <- melt(TodosEXP_TAXA,id = c("RastId","Kingdom","Name","SuperPhylum","Phylum","Class","Order","Family","Contigs","Size","RastNo"))
TodosHP.m<-subset(TodosHP.m,variable!="TOTAL")
#CyanosHeatPlot$RastId <- with(CyanosHeatPlot, reorder(CyanosHeatPlot$RastId, CyanosHeatPlot$TOTAL))
#CyanosHeatPlot.m <- melt(CyanosHeatPlot)
#CyanosHeatPlot.m<-subset(CyanosHeatPlot.m,variable!="TOTAL")
##   Boxplot expansiones en familias en colores por reinos
BoxCentralFamilies<-ggplot(TodosHP.m, aes(x=TodosHP.m$variable, y=TodosHP.m$value,color=TodosHP.m$Kingdom)) +labs(x = "Metabolic Families", y = "Copy number on Genomes",text = element_text(size=12)) + geom_boxplot()+theme_bw()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 10), legend.position = "bottom")
####ggsave(file="BoxCentralFamilies.svg", plot=BoxCentralFamilies, width=30, height=8)
############### End of boxplot   

######## 3) NOT READY Diversity of antiSMASH detection on 3 genomic DBs
CyanosSMASH.m <- melt(CyanosSMASH,id = c("RastId","Kingdom"))  
CyanosSMASH.m<-CyanosSMASH.m[(CyanosSMASH.m$value!=0)&(CyanosSMASH.m$value!="Cyanos")&(CyanosSMASH.m$variable!="TOTAL"),]
ArchaeasSMASH.m <- melt(ArchaeasSMASH,id = c("RastId","Kingdom"))  
ArchaeasSMASH.m<-ArchaeasSMASH.m[(ArchaeasSMASH.m$value!=0)&(ArchaeasSMASH.m$value!="Archea")&(ArchaeasSMASH.m$variable!="TOTAL"),]
ActinosSMASH.m <- melt(ActinosSMASH,id = c("RastId","Kingdom"))  
ActinosSMASH.m<-ActinosSMASH.m[(ActinosSMASH.m$value!=0)&(ActinosSMASH.m$value!="Actinos")&(ActinosSMASH.m$variable!="TOTAL"),]
Todos_SMASH<-rbind(ActinosSMASH.m,CyanosSMASH.m,ArchaeasSMASH.m)
DiversitySMASH<-ggplot2.barplot(data=Todos_SMASH, xName="Kingdom",  groupName="variable",showLegend=FALSE)
####ggsave(file="DiversitySMASH.svg", plot=DiversitySMASH, width=30, height=8)

## HACEr una funcion con la diversidad de Archaea 
#lista<-list(CyanosSMASH,ArchaeasSMASH,ActinosSMASH)
#lista<-list(NonZeroCyanos,NonZeroArchaeas,NonZeroActinos)
#TODOS_SMASH<-bind_rows(lista)
#aver<-TODOS_SMASH[TODOS_SMASH$RastId=="389046",]$TOTAL
#aver
#2 Tener boxplot de expansiones por familia por kingdom LISTOOO
#3 Escoger una familia para analizar a detalle por genero y poner sus árboles  PriA por ejemplo  

## Good code but not in use  
#names(CyanosSMASH)
#intersect(names(CyanosSMASH),names(ArchaeasSMASH))
#nonzero <- function(x) sum(x != 0)
#NonZeroCyanos<-numcolwise(nonzero)(CyanosSMASH)
#NonZeroArchaeas<-numcolwise(nonzero)(ArchaeasSMASH)
#NonZeroActinos<-numcolwise(nonzero)(ActinosSMASH)
#union( union(names(CyanosSMASH),names(ArchaeasSMASH)),names(ActinosSMASH))
#union( union(names(NonZeroCyanos),names(NonZeroArchaeas)),names(NonZeroActinos))

################## Init Figura antiSMASH propuesta por Paco Simil paper EvoMining  
# Previous                            #     Current      
#Genomic DB 230 genomas               #     Genomic DB   1245 genomes
# Central 106 families                #     Central 93 families 
# 226 BGCs                            #     1296 BGCs with 44552 enzymes
# Previous results            ######################################
# 101 expandidas                      #     93 Expansions, O bien 80 si consideramos solo expansiones on at least 1% (12.45) of organisms # Expandidas en coelicolor 33
# 23 enzimas con reclutamientos       #      82 ?? Recruitments  11 non recruitments 76 according to dataframe. 

## plan para replicar la figura CONSULTAR CON PACO Qué familias   
## Hacer un script que obtenga los BBH de las central con cada familia expandida 
## De los que nos son BBH cuales son antiSMASH /Cluster Finder  
## Cuales son MIBIG  
## Cuales no son nada.  

## I´m starting with Actinobacteria  
## 1) Identifying Familis with expansions  
## ActinosHeatPlot Has the hits    
 ## Now need a function how many organisms has expansions and in total how many expansions are from the 95 families  
#compare two vectors if ai > bi then 1 
ActinosHP <-ActinosHeatPlot
str(ActinosHP)
rownames(ActinosHP) <- ActinosHeatPlot[,1]
  HeatAverage=colMeans(ActinosHP)
  HeatSD=apply(ActinosHP, 2, sd)
ExpTresHold<-HeatAverage+HeatSD

### Coelicolor expansions   
Coelicolor<-ActinosHP["242137",]
names(Coelicolor)
Coelicolor<-Coelicolor[!names(Coelicolor) %in% c("TOTAL","RastId")]
ExpTresHold2<-ExpTresHold[!names(ExpTresHold) %in% c("TOTAL","RastId")]
CoeliExpansions<-Coelicolor>ExpTresHold2
CoeliExpansionsTotal<-sum(CoeliExpansions) ## how many expansions
CoeliExpansionsTotal ## 34
CoeliExpansionsFamilies<-Coelicolor[CoeliExpansions]## which families are expanded   
CoeliExpansionsFamilies
CoeliExpansionsFamilies<-names(Coelicolor)[CoeliExpansions]## which families are expanded   
CoeliExpansionsFamilies
########### End coelicolor expansions 

############## starting families exapnsions 
ExpTresHold$RastId<-"Treshold"

ActinosHP<-rbind (ActinosHP, ExpTresHold)  

FamiliesExpanded <- function(x){ ## is a column
  ultimo<-tail(x, n=1)
  subx<-x>ultimo ## vector of organisms with extra copies
  suma<-sum(subx) ## how many
return(suma)
}

ExpandedFamilies=apply(ActinosHP,2, FamiliesExpanded)  
ExpandedFamilies<-ExpandedFamilies[!names(ExpandedFamilies) %in% c("TOTAL","RastId")]
str(ExpandedFamilies)
nonzero <- function(x) sum(x != 0)
zero <- function(x) sum(x == 0)
Atleast1Percent<- function(x) sum(x >12.45)  

NonZeroExpandedFamilies<-nonzero(ExpandedFamilies)
ZeroExpandedFamilies<-zero(ExpandedFamilies)
Percent1ExpandedFamilies<-Atleast1Percent(ExpandedFamilies)
ZeroExpandedFamilies
NonZeroExpandedFamilies
Percent1ExpandedFamilies

ExpandedFamilies["Phosphoglycerate_dehydrogenase"]
#qplot(1:length(ExpandedFamilies), ExpandedFamilies, xlab ="Families") + geom_line()
qplot(names(ExpandedFamilies), ExpandedFamilies, xlab ="Families") + geom_line()+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 10), legend.position = "bottom")

### Ahora quien de estas familias tienen reclutamientos   
## Quiero grafica familia vs #reclutamientos 
ActinosNpReduced<-subset(ActinosNp,ActinosNp$np!="none")
cdata <- ddply(ActinosNpReduced, "Central", summarise,Nps  = length(np))
rownames(cdata) <- cdata[,1]
names(ExpandedFamilies)
row.names(cdata)

## Eliminar los 11 nones
ExpandedAndRecruited<-intersect(names(ExpandedFamilies),row.names(cdata))
cdata2<-subset(cdata,row.names(cdata) %in% ExpandedAndRecruited)
ExpandedFamilies2<-ExpandedFamilies[names(ExpandedFamilies) %in% ExpandedAndRecruited]
cdata2<-cdata2[match(names(ExpandedFamilies2), cdata2$Central),]
cdata2$Expansions=ExpandedFamilies2  
cdata2.m<- melt(cdata2,id = c("Central"))
##### Solo me falta graficar

ggplot(cdata2.m, aes(x = cdata2.m$Central, y = cdata2.m$value, fill = cdata2.m$variable))+ geom_bar(stat = "identity")+theme(plot.title = element_text(size = 14, face = "bold"), text = element_text(size = 12), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 10), legend.position = "bottom")

### ggplot2.barplot(data=cdata2.m, xName="Central",  groupName="value")
#cdata
#summary(ActinosNp)
## Quito none, interseccion y stack bar

#qplot(x=Central, data=ActinosNp, geom="bar", fill=np, position="stack")+theme_bw()+ theme(legend.position	="none",text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )

```


Expansions other central  
To acotate the search for enzymes of recent recruitment into natural products
  ![TauD](Figuras/TauDpseudomonas2.svg)  

As we have explored other genomic databases, central database is also customizable, on this case we have use TauD en enzyme from taurine asimilation sistem on enterobacteria, on the genomic database Pseudomonas. Enzymes does not necesary have to be central, in the sense that they belong to the core of its taxonomic group, but at least should have expansions.  

### Figure  3.1  Expansions on genomic dinamics  
3.2(Bakward EvoMining)    
Coelicolor clusters  
Esto se hará solo sobre Streptomyces de los 1246  

```{r CoelicolorMiBig, results = "asis"}
table <- read.csv("Figuras/CoelicolorMiBIG", row.names = 1,sep="\t")
kable(table,  caption = "Coelicolor\\label{tab:Coelicolor MiBig}",caption.short = "CoelicolorMiBig ")
```
     

```{r ExpNumber,warnings = FALSE, message = FALSE,echo=FALSE}
# Expansions of enzyme sequences from MiBIG from S coelicolor will be explored within the scope of the genomic database Streptomyces. The goal is to recover those enzymes that are not yet been considered as common on secondary metabolism. 
## Moda is the most common copy number on an organism, Organisms with an extra copy are the ones that may have this copy recrutied into secondary metabolism
# This extra copy on at least 4 organismos
## In addtiion the distribution of the enzyme is desired to be present on at least half o the organisms (Not to exclusive)
## Too exclusive means only belong to secondary metabolism, we are looking for switches  
## looking for an esay number between 0 and one that reflects too expanded, too exclusive
#  y el exp number,  mas entre .2<= Exp <=.6 y analizar eso árboles.  
# One minus average organisms that contains one copy.
# More copies than organisms this number tends to one   ## too expanded
# few copies  on homogeneously on few organisms tends to cero ## too particular  
# two copies by organism .5  , that is not usually the case  because there is some variance 


######### Functions   
OneOrMode <- function(x){ #max between 0 and one
a = table(x) # x is a vector
moda=a[which.max(a)]
inte=max(1,as.integer(names(moda)))
return(inte)
}

Mode <- function(x){ ## mode
a = table(x) # x is a vector
moda=a[which.max(a)]
inte=as.integer(names(moda))
return(inte)
}

OrganismsExtraCopy <- function(x){ ##how many organisms has an extra copy than the mode
  a = table(x) # x is a vector  
  moda=a[which.max(a)]
  inte=as.integer(names(moda)) #the moda
  subx<-as.integer(a[which(as.integer(names(a))>inte)]) ## vector of organisms with extra copies
  suma<-sum(subx) ## how many
return(suma)
}

OrgAtLeastOneCopy <- function(x){ ##how many organisms has a copy 
  a = table(x) # x is a vector  
  subx<-as.integer(a[which(as.integer(names(a))>0)]) ## vector of organnisms with at least one copy
  suma<-sum(subx) ## how many
return(suma)
}


Copies <- function(x){ ##how many organisms has an extra copy than the mode
  suma<-sum(x) ## how many
return(suma)
}

######## Reading and sorting data   
## Read EvoMining tables (head plots)
tableExp <- read.csv("Figuras/ExpansionBlast.data", header=TRUE, sep="\t")  ### Enzyme BGC Function from MiBIG coelicolor
tableCentral <- read.csv("Figuras/Central.data", header=TRUE, sep="\t")  ### Enzyme Subsystem Function from  
tableDistribution <- read.csv("Figuras/Enzymes.Distribution", header=TRUE, row.names=1,sep="\t")  ### Expansions from coelicolor (Heat plot score 0)
tableCentralDistribution <- read.csv("Figuras/CentralEnymes.Distribution.csv", header=TRUE, row.names=1,sep="\t") #Expansions from 93 enzimas central metabolism   

#necesito poner el valor de names de moda en el renglon con el mismo valor en enzima
#number of organisms greater than mode  at least tenpercent of the genome
##Reducing tableExp to those I have distribution
tableExp=tableExp[tableExp$Enzyme %in% names(tableDistribution),]
tableExp <- tableExp[order(tableExp$Enzyme),] 
tableDistribution <- tableDistribution[,order(names(tableDistribution))] 

tableExpOriginal <- read.csv("Figuras/ExpansionBlast.data", header=TRUE, sep="\t")  
tableExpOriginal=tableExp[tableExp$Enzyme %in% names(tableDistribution),]
tableExpOriginal <- tableExp[order(tableExp$Enzyme),] 

################# Processing functions BGC data  
modaOrOne=apply(tableDistribution,2, OneOrMode)  
moda=apply(tableDistribution,2, Mode)
ExtraCopy=apply(tableDistribution,2, OrganismsExtraCopy)
OneCopy=apply(tableDistribution,2, OrgAtLeastOneCopy)
CopiesEvo=apply(tableDistribution,2, Copies)
Average=colMeans(tableDistribution)
#names(moda)
tableExp$Moda=moda
tableExp$Average=Average
tableExp$ExtraCopy=ExtraCopy
tableExp$OneCopy=OneCopy
tableExp$CopiesEvo=CopiesEvo
#modaOrOne
#moda
#ExtraCopy
#OneCopy
CopiesEvo
#Average
#CopiesEvo
sum(tableDistribution$Enzyme_108)


tableExpOriginal$Moda=moda
tableExpOriginal$Average=Average
tableExpOriginal$ExtraCopy=ExtraCopy
tableExpOriginal$OneCopy=OneCopy
tableExpOriginal$CopiesEvo=CopiesEvo

################# Processing functions central data  
 CentralmodaOrOne=apply(tableCentralDistribution,2, OneOrMode)
Centralmoda=apply(tableCentralDistribution,2, Mode)
CentralExtraCopy=apply(tableCentralDistribution,2, OrganismsExtraCopy)
CentralOneCopy=apply(tableCentralDistribution,2, OrgAtLeastOneCopy)
CentralCopiesEvo=apply(tableCentralDistribution,2, Copies)
CentralAverage=colMeans(tableCentralDistribution)

tableCentral$Moda=Centralmoda
tableCentral$Average=CentralAverage
tableCentral$ExtraCopy=CentralExtraCopy
tableCentral$OneCopy=CentralOneCopy
tableCentral$CopiesEvo=CentralCopiesEvo

#One minus average organisms that contains one copy.
#More copies than organisms this number tends to one   ## too expanded
# few copies  on homogeneously on few organisms tends to cero ## too particular  
# two copies by organism   .5  , that is not usually the case  because there is some variance 
#tableExp$ExpNum=(modaOrOne-tableExp$OneCopy/tableExp$CopiesEvo)/(modaOrOne)  
#tableCentral$ExpNum=(CentralmodaOrOne-tableCentral$OneCopy/tableCentral$CopiesEvo)/(CentralmodaOrOne)  

tableExp$ExpNum=1-tableExp$OneCopy/tableExp$CopiesEvo
tableCentral$ExpNum=1-tableCentral$OneCopy/tableCentral$CopiesEvo


tableExp<-tableExp[order(tableExp$Order),]
#tableCentral$Order=tableCentral$Order+193
tableCentral$Order
tableCentral<-tableCentral[order(tableCentral$Order),]
tableEnzymeReduced=subset(tableExp,tableExp$ExpNum<=.6 & tableExp$ExpNum>= .2 )

total <- rbind(tableExp,tableCentral)
todosImage<-ggplot(total,aes(x=total$Order, y=total$ExpNum,  color=total$BGC, size = total$Moda))+facet_wrap(~ total$Type)+ geom_point(shape=20) + labs(x = "Metabolic Families", y = "Exp Number Actinobacteria Genomes",text = element_text(size=12)) + theme_bw()+theme(plot.title = element_text(size = 20, face = "bold"), text = element_text(size = 14), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 16), legend.position = "right")

ggplot(total,aes(x=total$Order, y=total$ExpNum,  color=total$BGC, size = total$Moda))+facet_wrap(~ total$Type)+ geom_point(shape=18) + labs(x = "Metabolic Families", y = "Exp Number Actinobacteria Genomes",text = element_text(size=12)) + theme_bw()+theme(plot.title = element_text(size = 20, face = "bold"), text = element_text(size = 14), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 16), legend.position = "right")

  ggsave(file="todosImage.svg", plot=todosImage, width=30, height=8)
  
#str(tableExp)
#tableExp$Enzyme
## ploting images
require("svglite")

#save the plot in a variable image to be able to export to svg
    imageBGC=ggplot(tableExp,aes(x=tableExp$Order, y=tableExp$ExpNum, color=tableExp$BGC, size = tableExp$Moda))+ geom_point() + labs(x = "Metabolic Families", y = "Exp Number Actinobacteria Genomes",text = element_text(size=12)) + theme_bw()+theme(plot.title = element_text(size = 20, face = "bold"), text = element_text(size = 14), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 16), legend.position = "bottom")
  ggsave(file="imageBGC.svg", plot=imageBGC, width=20, height=8)
  
  ggplot(tableExp,aes(x=tableExp$Order, y=tableExp$ExpNum, color=tableExp$BGC, size = tableExp$Moda))+ geom_point() + labs(x = "Metabolic Families", y = "Exp Number Actinobacteria Genomes",text = element_text(size=12)) + theme_bw()+theme(plot.title = element_text(size = 20, face = "bold"), text = element_text(size = 14), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 16), legend.position = "right")
    
  imageCentral= ggplot(tableCentral,aes(x=tableCentral$Order, y=tableCentral$ExpNum, color=tableCentral$BGC, size = tableCentral$Moda))+ geom_point() + labs(x = "Metabolic Families", y = "Exp Number Actinobacteria Genomes",text = element_text(size=12)) + theme_bw()+theme(plot.title = element_text(size = 20, face = "bold"), text = element_text(size = 14), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 16), legend.position = "right")
  ggsave(file="imageCentral.svg", plot=imageCentral, width=20, height=8)

    

    ggplot(tableEnzymeReduced,aes(x=tableEnzymeReduced$Order, y=tableEnzymeReduced$ExpNum, color=tableEnzymeReduced$BGC, size = tableEnzymeReduced$Moda))+ geom_point() + labs(x = "Metabolic Families", y = "Exp Number Actinobacteria Genomes",text = element_text(size=12)) + theme_bw()+theme(plot.title = element_text(size = 20, face = "bold"), text = element_text(size = 14), axis.title = element_text(face="bold"), axis.text.x=element_text(angle = 90,size = 16), legend.position = "right")

    
StreptoNp <- read.table("Figuras/naturalProductsEnzymeReduced.data", header=TRUE, sep="\t")


qplot(x=Central, data=StreptoNp, geom="bar", fill=kingdom, position="stack")+theme_bw()+ theme(legend.position	="bottom",text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )

qplot(x=Central, data=StreptoNp, geom="bar", fill=taxa, position="stack")+theme_bw()+ theme(legend.position	="bottom",text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )

mm<-count(StreptoNp$Central)
small<-mm[which(mm$freq<50),]
str(mm)

StreptoNpSmall=StreptoNp[StreptoNp$Central %in% small$x,]
StreptoNp$Central %in% small$x

qplot(x=Central, data=StreptoNpSmall, geom="bar", fill=kingdom, position="stack")+theme_bw()+ theme(legend.position	="bottom",text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )

qplot(x=Central, data=StreptoNpSmall, geom="bar", fill=taxa, position="stack")+theme_bw()+ theme(legend.position	="bottom",text = element_text(size=8), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5) ,axis.text.y = element_text(angle = 0, hjust = 1, vjust = 0.5) )


#kable(tableExp,  caption = "CoelicolorExpansions\\label{tab:Coelicolor Expansions}",caption.short = "CoelicolorExpansions")
```

Presence Absence
EvoMining was run over enzymes with expansion number between .1 and .6
Figuras/GenomicDBS.svg

## Methodology
Figure 1 Pipeline  

EvoMining algorithm and all its dependencies: blast, muscle, Gblocks, Quicktree and newick utilities were packaged on a docker container available at dockerhub, every container has inside EvoMining code which turns EvoMining in a totally open standalone tool, code at github. EvoMining analysis inputs are a genomic database (Genomic-DB), a database of seed family enzymes (Central-DB), and a natural product DB (np-DB), optionally antiSMASH analysis of the genomic DB may be used. Genomes on genomic-DB must be annotated by RAST, sequences of the central-DB and on np-DB must be on fasta format. MiBIG repository was used as default np-DB. Once databases are set, the first EvoMining step is to retrieve from the genomic-DB all sequences corresponding to the expanded families of the central-DB, to this end a blastp search (Default:e-value 0.001, score 100) with query seed enzymes from the central-DB against the genome-DB is conducted. Next, Best bidirectional hits between the expanded families from the previous step and enzymes

  starts with a blast search with query the CentralDB against the genomic DB,  comprises a docker  
s

this requirement provides functional annotation when looking for the genomic context of a particular enzyme.

Figure 2 Databases 
Three genomic database were integrated by 1245 Actinobacteria, 416 Cyanobacteria, 876 Archaea. Genomes were annotated by RAST.     
Central database were conformed by organisms X! X2 X3 on Actinobacteria Y1 Y2 Y· Cyanonbacteria, Z1 Z2 Z3 on Archaea 
Enzymes that are shared between this three taxonomical groups expansions were calculated, blast 0.0001, score 100 lenght bla bla
AntiSMASH version bla antiSMASH files, recruitment were run. 
r scripts at
Some trees were runned PriA 

Central database expanded to only central enzymas to shell metabolism, TauD was selected as an example, against XX Pseudomonas, it has a recruited 
Pseudomonas,
Figure 3 TauD  
TauD pseudomonas  
How where nps predicted  
All data base are available at Zenodo (Doi numbers genomic-DB: STREPTOMYCES,  Actinobacteria, Cyanobacteria, Pseudomonas, Archaea, central-DB:  Actinobacteria, Cyanobacteria, Pseudomonas, Archaea, lividans/coelicolor BGC ) 

Figure 4  Central vs Natural
Enzyme sequences from S. coelicolor and S. lividans byosynthetic gene clusters were retrieved from MiBiG to test how will behave as seeds from backward EvoMining  
Average copy number  

## Acknowledgments
Secretaria de Inovacion, Conabio:Keri/Ernesto, Argonne cluster

    [@dufresne_algorithmique_2016,@blin_recent_nodate,@kurtboke_revisiting_2017,@miller_interpreting_2017,@schniete_expanding_2017,@kim_recent_2017,@robertsen_toward_2017,@juarez-vazquez_evolution_nodate,@chavali_bioinformatics_nodate,@tracanna_mining_2017,@ren_breaking_2017,@choudhary_current_2017,@alanjary_antibiotic_2017,@chevrette_sandpuma:_2017,@wohlleben_antibiotic_2016,@weber_secondary_2016]
    
    

```{r bib, child = 'bibliography.Rmd'}
```
  